TestNames test flaky

```
cd ~/go-ssb/sbot
watch 'LIBRARIAN_WRITEALL=0 go test -v ./ -run TestNames | tee -a ~/testlog.txt'
```
Lots of PASSes.  But here's a FAIL:

```
2022/12/10 16:28:46 [margaret/indexes/badger] overwrote batch limit 0
=== RUN   TestNames
2022/12/10 16:28:46 saved identity @0CRrYbmaGxNBVRF3yPuLd0xQQyDIAT9GmIh+g/G9JSI=.ed25519 to testrun/TestNames/secrets/arny
2022/12/10 16:28:46 saved identity ssb:feed/gabbygrove-v1/YwZiHuC-3rhxLLg6jgHDe0u4E9r09La5NqcxQfMWghE= to testrun/TestNames/secrets/bert
2022/12/10 16:28:46 saved identity @izag8812GcTGDYE9ZkCbWo0M3HzcsxyeScsP/yDc0pU=.ed25519 to testrun/TestNames/secrets/cloe
2022/12/10 16:28:46 saved identity @YB33dgT4LHB97m7Du24bmiccAC1cX1MeMXofcCMU4kk=.ed25519 to testrun/TestNames/secret
    names_test.go:122: 
        	Error Trace:	/home/go-ssb/go-ssb/sbot/names_test.go:122
        	Error:      	Not equal: 
        	            	expected: "i'm cloe!"
        	            	actual  : "i'm arny!"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-i'm cloe!
        	            	+i'm arny!
        	Test:       	TestNames
        	Messages:   	did not the right name for cloe
level=debug event=update-replicate feed-want-count=0 hops=0 took=114.536µs
    leakcheck.go:113: Leaked goroutine: goroutine 10 [select]:
        github.com/dgraph-io/badger/v3.(*levelsController).runCompactor(0xc0022f0000, 0x1, 0xc00231a7e0)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:508 +0x30d
        created by github.com/dgraph-io/badger/v3.(*levelsController).startCompact
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:354 +0x4e
    leakcheck.go:113: Leaked goroutine: goroutine 11 [select]:
        github.com/dgraph-io/badger/v3.(*levelsController).runCompactor(0xc0022f0000, 0x2, 0xc00231a7e0)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:508 +0x30d
        created by github.com/dgraph-io/badger/v3.(*levelsController).startCompact
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:354 +0x4e
    leakcheck.go:113: Leaked goroutine: goroutine 12 [select]:
        github.com/dgraph-io/badger/v3.(*levelsController).runCompactor(0xc0022f0000, 0x3, 0xc00231a7e0)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:508 +0x30d
        created by github.com/dgraph-io/badger/v3.(*levelsController).startCompact
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:354 +0x4e
    leakcheck.go:113: Leaked goroutine: goroutine 13 [chan receive]:
        github.com/dgraph-io/badger/v3.(*DB).flushMemtable(0xc0000d5200, 0x0?)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:1078 +0xb2
        github.com/dgraph-io/badger/v3.Open.func5()
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:357 +0x25
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:356 +0x107c
    leakcheck.go:113: Leaked goroutine: goroutine 14 [select]:
        github.com/dgraph-io/badger/v3.(*vlogThreshold).listenForValueThresholdUpdate(0xc00221ccc0)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/value.go:1172 +0x11a
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:380 +0x16dc
    leakcheck.go:113: Leaked goroutine: goroutine 15 [select]:
        github.com/dgraph-io/badger/v3.(*DB).doWrites(0xc0000d5200, 0xc00231a8a0)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:900 +0x236
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:387 +0x179c
    leakcheck.go:113: Leaked goroutine: goroutine 16 [chan receive]:
        github.com/dgraph-io/badger/v3.(*valueLog).waitOnGC(0xc0000d53f8, 0xc00225c390?)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/value.go:1079 +0x7d
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:391 +0x185b
    leakcheck.go:113: Leaked goroutine: goroutine 23 [select]:
        github.com/ssbc/go-ssb/internal/ctxutils.WithError.func2()
        	/home/go-ssb/go-ssb/internal/ctxutils/withErr.go:31 +0x92
        created by github.com/ssbc/go-ssb/internal/ctxutils.WithError
        	/home/go-ssb/go-ssb/internal/ctxutils/withErr.go:30 +0x1df
    leakcheck.go:113: Leaked goroutine: goroutine 24 [chan receive]:
        github.com/ssbc/go-ssb/blobstore.(*WantManager).replicateLoop(0xc0000e4500)
        	/home/go-ssb/go-ssb/blobstore/wants.go:94 +0x57
        created by github.com/ssbc/go-ssb/blobstore.NewWantManager
        	/home/go-ssb/go-ssb/blobstore/wants.go:60 +0x30a
    leakcheck.go:113: Leaked goroutine: goroutine 25 [IO wait]:
        internal/poll.runtime_pollWait(0x7fe9e40c9218, 0x72)
        	/usr/lib/go-1.18/src/runtime/netpoll.go:302 +0x89
        internal/poll.(*pollDesc).wait(0xc0000ef100?, 0x203000?, 0x0)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:83 +0x32
        internal/poll.(*pollDesc).waitRead(...)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:88
        internal/poll.(*FD).Accept(0xc0000ef100)
        	/usr/lib/go-1.18/src/internal/poll/fd_unix.go:614 +0x22c
        net.(*netFD).accept(0xc0000ef100)
        	/usr/lib/go-1.18/src/net/fd_unix.go:172 +0x35
        net.(*UnixListener).accept(0x4433c0?)
        	/usr/lib/go-1.18/src/net/unixsock_posix.go:166 +0x1c
        net.(*UnixListener).Accept(0xc0021fbb60)
        	/usr/lib/go-1.18/src/net/unixsock.go:260 +0x3d
        github.com/ssbc/go-ssb/sbot.unixSockServer.accept({{0xffc3b0, 0xc0021fb7d0}, {0xff5ca0, 0xc0021f3d50}, {0xffb0a0, 0xc0021fbb60}, {0x0, 0x0, 0x0}, {0xc0021fbb90, ...}, ...})
        	/home/go-ssb/go-ssb/sbot/options.go:175 +0x64
        created by github.com/ssbc/go-ssb/sbot.WithUNIXSocket.func1
        	/home/go-ssb/go-ssb/sbot/options.go:153 +0x33b
    leakcheck.go:113: Leaked goroutine: goroutine 26 [select]:
        github.com/ssbc/margaret/offset2.(*offsetQuery).Push(0xc000090480, {0xffc3b0, 0xc0021fb7d0}, {0x7fe9e4084558?, 0xc0000102a8?})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/offset2/qry.go:202 +0xc9
        github.com/ssbc/go-luigi.Pump({0xffc3b0, 0xc0021fb7d0}, {0x7fe9e4084558?, 0xc0000102a8?}, {0xff5ba0, 0xc000090480})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:56 +0x82
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 27 [select]:
        github.com/dgraph-io/badger/v3/y.(*WaterMark).process(0xc00225c3c0, 0xc00225c390)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/y/watermark.go:214 +0x285
        created by github.com/dgraph-io/badger/v3/y.(*WaterMark).Init
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/y/watermark.go:72 +0xaa
    leakcheck.go:113: Leaked goroutine: goroutine 28 [select]:
        github.com/dgraph-io/badger/v3/y.(*WaterMark).process(0xc00225c3f0, 0xc00225c390)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/y/watermark.go:214 +0x285
        created by github.com/dgraph-io/badger/v3/y.(*WaterMark).Init
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/y/watermark.go:72 +0xaa
    leakcheck.go:113: Leaked goroutine: goroutine 29 [select]:
        github.com/dgraph-io/ristretto/z.(*AllocatorPool).freeupAllocators(0xc00000f5f0)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/z/allocator.go:385 +0x150
        created by github.com/dgraph-io/ristretto/z.NewAllocatorPool
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/z/allocator.go:324 +0xc5
    leakcheck.go:113: Leaked goroutine: goroutine 30 [select]:
        github.com/dgraph-io/ristretto.(*defaultPolicy).processItems(0xc00221cd40)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/policy.go:102 +0x91
        created by github.com/dgraph-io/ristretto.newDefaultPolicy
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/policy.go:86 +0x156
    leakcheck.go:113: Leaked goroutine: goroutine 31 [select]:
        github.com/dgraph-io/ristretto.(*Cache).processItems(0xc0000ef580)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/cache.go:452 +0x15e
        created by github.com/dgraph-io/ristretto.NewCache
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/cache.go:207 +0x696
    leakcheck.go:113: Leaked goroutine: goroutine 32 [select]:
        github.com/dgraph-io/badger/v3.(*DB).monitorCache(0xc0000d5200, 0xc00231a660)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:469 +0x18a
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:311 +0xc8b
    leakcheck.go:113: Leaked goroutine: goroutine 33 [select]:
        github.com/dgraph-io/badger/v3.(*DB).updateSize(0xc0000d5200, 0xc00231a7b0)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:1171 +0x158
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:331 +0xe8c
    leakcheck.go:113: Leaked goroutine: goroutine 34 [select]:
        github.com/dgraph-io/badger/v3.(*publisher).listenForUpdates(0xc00225c4b0, 0xc00231a900)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/publisher.go:73 +0x150
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:395 +0x190a
    leakcheck.go:113: Leaked goroutine: goroutine 35 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc0001221e0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 36 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc0001222d0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 37 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc000122370)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 38 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc000122410)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 39 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc0000e4780)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 40 [select]:
        github.com/ssbc/margaret/offset2.(*offsetQuery).Push(0xc000033320, {0xffc3b0, 0xc0021fb7d0}, {0x7fe9ddbc9878?, 0xc00000e468?})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/offset2/qry.go:202 +0xc9
        github.com/ssbc/go-luigi.Pump({0xffc3b0, 0xc0021fb7d0}, {0x7fe9ddbc9878?, 0xc00000e468?}, {0xff5ba0, 0xc000033320})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:56 +0x82
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 41 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc0000e4820)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 42 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc0001227d0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 43 [select]:
        github.com/ssbc/margaret/offset2.(*offsetQuery).Push(0xc00232e120, {0xffc3b0, 0xc0021fb7d0}, {0x7fe9ddbcb198?, 0xc0000c0000?})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/offset2/qry.go:202 +0xc9
        github.com/ssbc/go-luigi.Pump({0xffc3b0, 0xc0021fb7d0}, {0x7fe9ddbcb198?, 0xc0000c0000?}, {0xff5ba0, 0xc00232e120})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:56 +0x82
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 44 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc0000e48c0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 45 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*query).livequery(0xc0000be5d0, {0xffc3b0, 0xc0021fb7d0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:158 +0xe7
        github.com/ssbc/margaret/multilog/roaring.(*query).Next(0xc0000be5d0, {0xffc3b0, 0xc0021fb7d0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:126 +0x328
        github.com/ssbc/go-luigi/mfr.(*srcMap).Next(0xc0000c85d0, {0xffc3b0, 0xc0021fb7d0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/mfr/map.go:58 +0x33
        github.com/ssbc/go-luigi.Pump({0xffc3b0, 0xc0021fb7d0}, {0x7fe9ddbc9878?, 0xc00000ed38?}, {0xff5680, 0xc0000c85d0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:60 +0xac
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 46 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc0000e4a00)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 47 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*query).livequery(0xc0000beff0, {0xffc3b0, 0xc0021fb7d0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:158 +0xe7
        github.com/ssbc/margaret/multilog/roaring.(*query).Next(0xc0000beff0, {0xffc3b0, 0xc0021fb7d0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:126 +0x328
        github.com/ssbc/go-luigi/mfr.(*srcMap).Next(0xc0000c8768, {0xffc3b0, 0xc0021fb7d0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/mfr/map.go:58 +0x33
        github.com/ssbc/go-luigi.Pump({0xffc3b0, 0xc0021fb7d0}, {0x7fe9ddbc9878?, 0xc00000edc8?}, {0xff5680, 0xc0000c8768})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:60 +0xac
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 48 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc0000e4be0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 49 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*query).livequery(0xc0000bf2f0, {0xffc3b0, 0xc0021fb7d0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:158 +0xe7
        github.com/ssbc/margaret/multilog/roaring.(*query).Next(0xc0000bf2f0, {0xffc3b0, 0xc0021fb7d0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:126 +0x328
        github.com/ssbc/go-luigi/mfr.(*srcMap).Next(0xc0000c8840, {0xffc3b0, 0xc0021fb7d0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/mfr/map.go:58 +0x33
        github.com/ssbc/go-luigi.Pump({0xffc3b0, 0xc0021fb7d0}, {0x7fe9ddbc9878?, 0xc00000ee88?}, {0xff5680, 0xc0000c8840})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:60 +0xac
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 66 [select]:
        github.com/ssbc/go-ssb/sbot.debounce({0xffc3b0, 0xc0021fb7d0}, 0xb2d05e00, {0xff9f90, 0xc00221ca00}, 0xc000091860)
        	/home/go-ssb/go-ssb/sbot/replicate.go:135 +0x16c
        created by github.com/ssbc/go-ssb/sbot.(*Sbot).newGraphReplicator
        	/home/go-ssb/go-ssb/sbot/replicate.go:73 +0x3c5
    leakcheck.go:113: Leaked goroutine: goroutine 67 [select]:
        github.com/ssbc/margaret/offset2.(*offsetQuery).Push(0xc00232e960, {0xffc3b0, 0xc0021fb7d0}, {0xff9350?, 0xc00008b060?})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/offset2/qry.go:202 +0xc9
        github.com/ssbc/go-luigi.Pump({0xffc3b0, 0xc0021fb7d0}, {0xff9350?, 0xc00008b060?}, {0xff5ba0, 0xc00232e960})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:56 +0x82
        github.com/ssbc/go-ssb/plugins/gossip.(*FeedManager).serveLiveFeeds(0xc00220e3f0)
        	/home/go-ssb/go-ssb/plugins/gossip/feed_manager.go:106 +0x168
        created by github.com/ssbc/go-ssb/plugins/gossip.NewFeedManager
        	/home/go-ssb/go-ssb/plugins/gossip/feed_manager.go:69 +0x1ea
    leakcheck.go:113: Leaked goroutine: goroutine 9 [select]:
        github.com/dgraph-io/badger/v3.(*levelsController).runCompactor(0xc0022f0000, 0x0, 0xc00231a7e0)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:508 +0x30d
        created by github.com/dgraph-io/badger/v3.(*levelsController).startCompact
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:354 +0x4e
    leakcheck.go:113: Leaked goroutine: goroutine 92 [IO wait]:
        internal/poll.runtime_pollWait(0x7fe9e40c9128, 0x72)
        	/usr/lib/go-1.18/src/runtime/netpoll.go:302 +0x89
        internal/poll.(*pollDesc).wait(0xc00232d480?, 0xc00220b580?, 0x0)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:83 +0x32
        internal/poll.(*pollDesc).waitRead(...)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:88
        internal/poll.(*FD).Read(0xc00232d480, {0xc00220b580, 0x9, 0x9})
        	/usr/lib/go-1.18/src/internal/poll/fd_unix.go:167 +0x25a
        net.(*netFD).Read(0xc00232d480, {0xc00220b580?, 0x10000005423fc?, 0x7fe9e40bd318?})
        	/usr/lib/go-1.18/src/net/fd_posix.go:55 +0x29
        net.(*conn).Read(0xc0000cc520, {0xc00220b580?, 0xc00233faf8?, 0x44fef2?})
        	/usr/lib/go-1.18/src/net/net.go:183 +0x45
        io.ReadAtLeast({0xff6400, 0xc0000cc520}, {0xc00220b580, 0x9, 0x9}, 0x9)
        	/usr/lib/go-1.18/src/io/io.go:331 +0x9a
        io.ReadFull(...)
        	/usr/lib/go-1.18/src/io/io.go:350
        encoding/binary.Read({0xff6400, 0xc0000cc520}, {0xfff1c8, 0x15972e8}, {0xd3a9e0?, 0xc00220b574})
        	/usr/lib/go-1.18/src/encoding/binary/binary.go:256 +0xc3e
        github.com/ssbc/go-muxrpc/v2/codec.Reader.ReadHeader({{0xff6400?, 0xc0000cc520?}}, 0xc00220b574)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/codec/reader.go:48 +0x45
        github.com/ssbc/go-muxrpc/v2.(*Packer).NextHeader(0xc00232ff20, {0xffc030, 0xc00227e580}, 0xc00220b574)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/packer.go:49 +0xcb
        github.com/ssbc/go-muxrpc/v2.(*rpc).serve.func2(0xc000109ba0, 0xffc030?, 0xc00233fef8)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:359 +0x5e
        github.com/ssbc/go-muxrpc/v2.(*rpc).serve(0xc000109ba0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:378 +0x19c
        github.com/ssbc/go-muxrpc/v2.Handle.func2()
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:128 +0x25
        created by github.com/ssbc/go-muxrpc/v2.Handle
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:127 +0x536
    leakcheck.go:113: Leaked goroutine: goroutine 93 [chan receive]:
        github.com/ssbc/go-muxrpc/v2.(*rpc).Serve(0xc00228e240?)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:336 +0x2b
        github.com/ssbc/go-ssb/sbot.unixSockServer.serve({{0xffc3b0, 0xc0021fb7d0}, {0xff5ca0, 0xc0021f3d50}, {0xffb0a0, 0xc0021fbb60}, {0x0, 0x0, 0x0}, {0xc0021fbb90, ...}, ...}, ...)
        	/home/go-ssb/go-ssb/sbot/options.go:228 +0x4f9
        created by github.com/ssbc/go-ssb/sbot.unixSockServer.accept
        	/home/go-ssb/go-ssb/sbot/options.go:205 +0x51f
    leakcheck.go:113: Leaked goroutine: goroutine 94 [select]:
        context.propagateCancel.func1()
        	/usr/lib/go-1.18/src/context/context.go:279 +0x9e
        created by context.propagateCancel
        	/usr/lib/go-1.18/src/context/context.go:278 +0x1cb
    leakcheck.go:113: Leaked goroutine: goroutine 96 [IO wait]:
        internal/poll.runtime_pollWait(0x7fe9e40c9038, 0x72)
        	/usr/lib/go-1.18/src/runtime/netpoll.go:302 +0x89
        internal/poll.(*pollDesc).wait(0xc00232d880?, 0xc0022987e0?, 0x0)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:83 +0x32
        internal/poll.(*pollDesc).waitRead(...)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:88
        internal/poll.(*FD).Read(0xc00232d880, {0xc0022987e0, 0x9, 0x9})
        	/usr/lib/go-1.18/src/internal/poll/fd_unix.go:167 +0x25a
        net.(*netFD).Read(0xc00232d880, {0xc0022987e0?, 0x38?, 0x40f845?})
        	/usr/lib/go-1.18/src/net/fd_posix.go:55 +0x29
        net.(*conn).Read(0xc0000cc538, {0xc0022987e0?, 0x7fea0b6ed5b8?, 0x10?})
        	/usr/lib/go-1.18/src/net/net.go:183 +0x45
        io.ReadAtLeast({0x7fe9ddbcfd18, 0xc0000752c0}, {0xc0022987e0, 0x9, 0x9}, 0x9)
        	/usr/lib/go-1.18/src/io/io.go:331 +0x9a
        io.ReadFull(...)
        	/usr/lib/go-1.18/src/io/io.go:350
        encoding/binary.Read({0x7fe9ddbcfd18, 0xc0000752c0}, {0xfff1c8, 0x15972e8}, {0xd3a9e0?, 0xc0022987d0})
        	/usr/lib/go-1.18/src/encoding/binary/binary.go:256 +0xc3e
        github.com/ssbc/go-muxrpc/v2/codec.Reader.ReadHeader({{0x7fe9ddbcfd18?, 0xc0000752c0?}}, 0xc0022987d0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/codec/reader.go:48 +0x45
        github.com/ssbc/go-muxrpc/v2.(*Packer).NextHeader(0xc00228e240, {0xffc030, 0xc00227e680}, 0xc0022987d0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/packer.go:49 +0xcb
        github.com/ssbc/go-muxrpc/v2.(*rpc).serve.func2(0xc002234000, 0xffc030?, 0xc002345ef8)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:359 +0x5e
        github.com/ssbc/go-muxrpc/v2.(*rpc).serve(0xc002234000)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:378 +0x19c
        github.com/ssbc/go-muxrpc/v2.Handle.func2()
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:128 +0x25
        created by github.com/ssbc/go-muxrpc/v2.Handle
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:127 +0x536
    leakcheck.go:113: Leaked goroutine: goroutine 99 [chan receive]:
        github.com/ssbc/go-muxrpc/v2.(*rpc).Serve(0xc00017b7d0?)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:336 +0x2b
        github.com/ssbc/go-ssb/client.NewUnix.func1()
        	/home/go-ssb/go-ssb/client/client.go:169 +0x4f
        created by github.com/ssbc/go-ssb/client.NewUnix
        	/home/go-ssb/go-ssb/client/client.go:168 +0x425
--- FAIL: TestNames (10.09s)
FAIL
```

So, after making this change:

```
@@ -114,18 +114,18 @@ func TestNames(t *testing.T) {
                "cloe": "i'm cloe!",
        }
 
-       r.Len(all, len(want), "expected entries for all three keypairs")
+       r.Len(all, len(want), "expected entries for all three keypairs but received %#v", all)
```

I get:

```
2022/12/10 17:09:02 [margaret/indexes/badger] overwrote batch limit 0
=== RUN   TestNames
2022/12/10 17:09:02 saved identity @4rgZC2P3Swg9bpPu1/dJ6DCnYVPoJrf/knuFuhUx4dw=.ed25519 to testrun/TestNames/secrets/arny
2022/12/10 17:09:02 saved identity ssb:feed/gabbygrove-v1/Sl9TMhc_enkCdCxabZ6IcP96iVTqAQtv9czosAeFLc0= to testrun/TestNames/secrets/bert
2022/12/10 17:09:02 saved identity @xR+INBvqdeR1wJf4F6B6TPXLIi3kxc0sJxRFxNK+IvE=.ed25519 to testrun/TestNames/secrets/cloe
2022/12/10 17:09:02 saved identity @h5cuxpsVLvnj9fA3JYz9APSMEmCu6IXP7ERVL7OpqhI=.ed25519 to testrun/TestNames/secret
    names_test.go:117: 
        	Error Trace:	/home/go-ssb/go-ssb/sbot/names_test.go:117
        	Error:      	"map[@4rgZC2P3Swg9bpPu1/dJ6DCnYVPoJrf/knuFuhUx4dw=.ed25519:map[@4rgZC2P3Swg9bpPu1/dJ6DCnYVPoJrf/knuFuhUx4dw=.ed25519:i'm arny!]]" should have 3 item(s), but has 1
        	Test:       	TestNames
        	Messages:   	expected entries for all three keypairs but received client.NamesGetResult{"@4rgZC2P3Swg9bpPu1/dJ6DCnYVPoJrf/knuFuhUx4dw=.ed25519":map[string]string{"@4rgZC2P3Swg9bpPu1/dJ6DCnYVPoJrf/knuFuhUx4dw=.ed25519":"i'm arny!"}}
level=debug event=update-replicate feed-want-count=0 hops=0 took=210.415µs
    leakcheck.go:113: Leaked goroutine: goroutine 10 [select]:
        github.com/ssbc/go-ssb/internal/ctxutils.WithError.func2()
        	/home/go-ssb/go-ssb/internal/ctxutils/withErr.go:31 +0x92
        created by github.com/ssbc/go-ssb/internal/ctxutils.WithError
        	/home/go-ssb/go-ssb/internal/ctxutils/withErr.go:30 +0x1df
    leakcheck.go:113: Leaked goroutine: goroutine 11 [chan receive]:
        github.com/ssbc/go-ssb/blobstore.(*WantManager).replicateLoop(0xc00014bae0)
        	/home/go-ssb/go-ssb/blobstore/wants.go:94 +0x57
        created by github.com/ssbc/go-ssb/blobstore.NewWantManager
        	/home/go-ssb/go-ssb/blobstore/wants.go:60 +0x30a
    leakcheck.go:113: Leaked goroutine: goroutine 12 [IO wait]:
        internal/poll.runtime_pollWait(0x7fac09550f58, 0x72)
        	/usr/lib/go-1.18/src/runtime/netpoll.go:302 +0x89
        internal/poll.(*pollDesc).wait(0xc000159100?, 0x203000?, 0x0)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:83 +0x32
        internal/poll.(*pollDesc).waitRead(...)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:88
        internal/poll.(*FD).Accept(0xc000159100)
        	/usr/lib/go-1.18/src/internal/poll/fd_unix.go:614 +0x22c
        net.(*netFD).accept(0xc000159100)
        	/usr/lib/go-1.18/src/net/fd_unix.go:172 +0x35
        net.(*UnixListener).accept(0x4433c0?)
        	/usr/lib/go-1.18/src/net/unixsock_posix.go:166 +0x1c
        net.(*UnixListener).Accept(0xc0000fdce0)
        	/usr/lib/go-1.18/src/net/unixsock.go:260 +0x3d
        github.com/ssbc/go-ssb/sbot.unixSockServer.accept({{0xffc450, 0xc0000fd950}, {0xff5d40, 0xc0021f3ea0}, {0xffb140, 0xc0000fdce0}, {0x0, 0x0, 0x0}, {0xc0000fdd10, ...}, ...})
        	/home/go-ssb/go-ssb/sbot/options.go:175 +0x64
        created by github.com/ssbc/go-ssb/sbot.WithUNIXSocket.func1
        	/home/go-ssb/go-ssb/sbot/options.go:153 +0x33b
    leakcheck.go:113: Leaked goroutine: goroutine 13 [select]:
        github.com/ssbc/margaret/offset2.(*offsetQuery).Push(0xc0000324e0, {0xffc450, 0xc0000fd950}, {0x7fac10554a18?, 0xc000130748?})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/offset2/qry.go:202 +0xc9
        github.com/ssbc/go-luigi.Pump({0xffc450, 0xc0000fd950}, {0x7fac10554a18?, 0xc000130748?}, {0xff5c40, 0xc0000324e0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:56 +0x82
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 14 [select]:
        github.com/dgraph-io/badger/v3/y.(*WaterMark).process(0xc00224e540, 0xc00224e510)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/y/watermark.go:214 +0x285
        created by github.com/dgraph-io/badger/v3/y.(*WaterMark).Init
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/y/watermark.go:72 +0xaa
    leakcheck.go:113: Leaked goroutine: goroutine 15 [select]:
        github.com/dgraph-io/badger/v3/y.(*WaterMark).process(0xc00224e570, 0xc00224e510)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/y/watermark.go:214 +0x285
        created by github.com/dgraph-io/badger/v3/y.(*WaterMark).Init
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/y/watermark.go:72 +0xaa
    leakcheck.go:113: Leaked goroutine: goroutine 16 [select]:
        github.com/dgraph-io/ristretto/z.(*AllocatorPool).freeupAllocators(0xc00012b890)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/z/allocator.go:385 +0x150
        created by github.com/dgraph-io/ristretto/z.NewAllocatorPool
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/z/allocator.go:324 +0xc5
    leakcheck.go:113: Leaked goroutine: goroutine 22 [select]:
        github.com/dgraph-io/badger/v3.(*levelsController).runCompactor(0xc002210070, 0x0, 0xc0025009c0)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:508 +0x30d
        created by github.com/dgraph-io/badger/v3.(*levelsController).startCompact
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:354 +0x4e
    leakcheck.go:113: Leaked goroutine: goroutine 23 [select]:
        github.com/dgraph-io/badger/v3.(*levelsController).runCompactor(0xc002210070, 0x1, 0xc0025009c0)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:508 +0x30d
        created by github.com/dgraph-io/badger/v3.(*levelsController).startCompact
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:354 +0x4e
    leakcheck.go:113: Leaked goroutine: goroutine 24 [select]:
        github.com/dgraph-io/badger/v3.(*levelsController).runCompactor(0xc002210070, 0x2, 0xc0025009c0)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:508 +0x30d
        created by github.com/dgraph-io/badger/v3.(*levelsController).startCompact
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:354 +0x4e
    leakcheck.go:113: Leaked goroutine: goroutine 25 [select]:
        github.com/dgraph-io/badger/v3.(*levelsController).runCompactor(0xc002210070, 0x3, 0xc0025009c0)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:508 +0x30d
        created by github.com/dgraph-io/badger/v3.(*levelsController).startCompact
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:354 +0x4e
    leakcheck.go:113: Leaked goroutine: goroutine 26 [chan receive]:
        github.com/dgraph-io/badger/v3.(*DB).flushMemtable(0xc0000e7200, 0x0?)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:1078 +0xb2
        github.com/dgraph-io/badger/v3.Open.func5()
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:357 +0x25
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:356 +0x107c
    leakcheck.go:113: Leaked goroutine: goroutine 27 [select]:
        github.com/dgraph-io/badger/v3.(*vlogThreshold).listenForValueThresholdUpdate(0xc000081bc0)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/value.go:1172 +0x11a
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:380 +0x16dc
    leakcheck.go:113: Leaked goroutine: goroutine 28 [select]:
        github.com/dgraph-io/badger/v3.(*DB).doWrites(0xc0000e7200, 0xc002500a80)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:900 +0x236
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:387 +0x179c
    leakcheck.go:113: Leaked goroutine: goroutine 29 [chan receive]:
        github.com/dgraph-io/badger/v3.(*valueLog).waitOnGC(0xc0000e73f8, 0x0?)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/value.go:1079 +0x7d
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:391 +0x185b
    leakcheck.go:113: Leaked goroutine: goroutine 30 [select]:
        github.com/dgraph-io/badger/v3.(*publisher).listenForUpdates(0xc00224e630, 0xc002500ae0)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/publisher.go:73 +0x150
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:395 +0x190a
    leakcheck.go:113: Leaked goroutine: goroutine 31 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc000154190)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 32 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc000154280)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 33 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc000154320)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 34 [select]:
        github.com/dgraph-io/ristretto.(*defaultPolicy).processItems(0xc000081c40)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/policy.go:102 +0x91
        created by github.com/dgraph-io/ristretto.newDefaultPolicy
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/policy.go:86 +0x156
    leakcheck.go:113: Leaked goroutine: goroutine 35 [select]:
        github.com/dgraph-io/ristretto.(*Cache).processItems(0xc000159580)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/cache.go:452 +0x15e
        created by github.com/dgraph-io/ristretto.NewCache
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/cache.go:207 +0x696
    leakcheck.go:113: Leaked goroutine: goroutine 36 [select]:
        github.com/dgraph-io/badger/v3.(*DB).monitorCache(0xc0000e7200, 0xc0025007e0)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:469 +0x18a
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:311 +0xc8b
    leakcheck.go:113: Leaked goroutine: goroutine 37 [select]:
        github.com/dgraph-io/badger/v3.(*DB).updateSize(0xc0000e7200, 0xc002500930)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:1171 +0x158
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:331 +0xe8c
    leakcheck.go:113: Leaked goroutine: goroutine 50 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc0001543c0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 51 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc00014a8c0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 52 [select]:
        github.com/ssbc/margaret/offset2.(*offsetQuery).Push(0xc0022ac060, {0xffc450, 0xc0000fd950}, {0x7fac1002dbb8?, 0xc00012a600?})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/offset2/qry.go:202 +0xc9
        github.com/ssbc/go-luigi.Pump({0xffc450, 0xc0000fd950}, {0x7fac1002dbb8?, 0xc00012a600?}, {0xff5c40, 0xc0022ac060})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:56 +0x82
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 53 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc00014ab40)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 54 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc000154780)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 55 [select]:
        github.com/ssbc/margaret/offset2.(*offsetQuery).Push(0xc0025134a0, {0xffc450, 0xc0000fd950}, {0x7fac1002dc60?, 0xc000122000?})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/offset2/qry.go:202 +0xc9
        github.com/ssbc/go-luigi.Pump({0xffc450, 0xc0000fd950}, {0x7fac1002dc60?, 0xc000122000?}, {0xff5c40, 0xc0025134a0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:56 +0x82
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 56 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc00014abe0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 57 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*query).livequery(0xc00223a270, {0xffc450, 0xc0000fd950})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:158 +0xe7
        github.com/ssbc/margaret/multilog/roaring.(*query).Next(0xc00223a270, {0xffc450, 0xc0000fd950})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:126 +0x328
        github.com/ssbc/go-luigi/mfr.(*srcMap).Next(0xc00000f788, {0xffc450, 0xc0000fd950})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/mfr/map.go:58 +0x33
        github.com/ssbc/go-luigi.Pump({0xffc450, 0xc0000fd950}, {0x7fac1002dbb8?, 0xc00012a6d8?}, {0xff5720, 0xc00000f788})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:60 +0xac
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 58 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc00014ae60)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 59 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*query).livequery(0xc00223a390, {0xffc450, 0xc0000fd950})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:158 +0xe7
        github.com/ssbc/margaret/multilog/roaring.(*query).Next(0xc00223a390, {0xffc450, 0xc0000fd950})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:126 +0x328
        github.com/ssbc/go-luigi/mfr.(*srcMap).Next(0xc00000f860, {0xffc450, 0xc0000fd950})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/mfr/map.go:58 +0x33
        github.com/ssbc/go-luigi.Pump({0xffc450, 0xc0000fd950}, {0x7fac1002dbb8?, 0xc00012aed0?}, {0xff5720, 0xc00000f860})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:60 +0xac
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 60 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc00014b220)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 61 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*query).livequery(0xc00223a4b0, {0xffc450, 0xc0000fd950})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:158 +0xe7
        github.com/ssbc/margaret/multilog/roaring.(*query).Next(0xc00223a4b0, {0xffc450, 0xc0000fd950})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:126 +0x328
        github.com/ssbc/go-luigi/mfr.(*srcMap).Next(0xc00000f938, {0xffc450, 0xc0000fd950})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/mfr/map.go:58 +0x33
        github.com/ssbc/go-luigi.Pump({0xffc450, 0xc0000fd950}, {0x7fac1002dbb8?, 0xc00012b020?}, {0xff5720, 0xc00000f938})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:60 +0xac
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 62 [select]:
        github.com/ssbc/go-ssb/sbot.debounce({0xffc450, 0xc0000fd950}, 0xb2d05e00, {0xffa030, 0xc000081900}, 0xc000033800)
        	/home/go-ssb/go-ssb/sbot/replicate.go:135 +0x16c
        created by github.com/ssbc/go-ssb/sbot.(*Sbot).newGraphReplicator
        	/home/go-ssb/go-ssb/sbot/replicate.go:73 +0x3c5
    leakcheck.go:113: Leaked goroutine: goroutine 63 [select]:
        github.com/ssbc/margaret/offset2.(*offsetQuery).Push(0xc002513800, {0xffc450, 0xc0000fd950}, {0xff93f0?, 0xc0021f2750?})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/offset2/qry.go:202 +0xc9
        github.com/ssbc/go-luigi.Pump({0xffc450, 0xc0000fd950}, {0xff93f0?, 0xc0021f2750?}, {0xff5c40, 0xc002513800})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:56 +0x82
        github.com/ssbc/go-ssb/plugins/gossip.(*FeedManager).serveLiveFeeds(0xc002210460)
        	/home/go-ssb/go-ssb/plugins/gossip/feed_manager.go:106 +0x168
        created by github.com/ssbc/go-ssb/plugins/gossip.NewFeedManager
        	/home/go-ssb/go-ssb/plugins/gossip/feed_manager.go:69 +0x1ea
    leakcheck.go:113: Leaked goroutine: goroutine 65 [IO wait]:
        internal/poll.runtime_pollWait(0x7fac09550e68, 0x72)
        	/usr/lib/go-1.18/src/runtime/netpoll.go:302 +0x89
        internal/poll.(*pollDesc).wait(0xc002258000?, 0xc00220c8f0?, 0x0)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:83 +0x32
        internal/poll.(*pollDesc).waitRead(...)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:88
        internal/poll.(*FD).Read(0xc002258000, {0xc00220c8f0, 0x9, 0x9})
        	/usr/lib/go-1.18/src/internal/poll/fd_unix.go:167 +0x25a
        net.(*netFD).Read(0xc002258000, {0xc00220c8f0?, 0x10000005423fc?, 0x7fac1051eed8?})
        	/usr/lib/go-1.18/src/net/fd_posix.go:55 +0x29
        net.(*conn).Read(0xc0001306f8, {0xc00220c8f0?, 0xc002533af8?, 0x44fef2?})
        	/usr/lib/go-1.18/src/net/net.go:183 +0x45
        io.ReadAtLeast({0xff64a0, 0xc0001306f8}, {0xc00220c8f0, 0x9, 0x9}, 0x9)
        	/usr/lib/go-1.18/src/io/io.go:331 +0x9a
        io.ReadFull(...)
        	/usr/lib/go-1.18/src/io/io.go:350
        encoding/binary.Read({0xff64a0, 0xc0001306f8}, {0xfff268, 0x15972e8}, {0xd3a9e0?, 0xc00220c8e0})
        	/usr/lib/go-1.18/src/encoding/binary/binary.go:256 +0xc3e
        github.com/ssbc/go-muxrpc/v2/codec.Reader.ReadHeader({{0xff64a0?, 0xc0001306f8?}}, 0xc00220c8e0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/codec/reader.go:48 +0x45
        github.com/ssbc/go-muxrpc/v2.(*Packer).NextHeader(0xc0001018c0, {0xffc0d0, 0xc0021c3f40}, 0xc00220c8e0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/packer.go:49 +0xcb
        github.com/ssbc/go-muxrpc/v2.(*rpc).serve.func2(0xc0001324e0, 0xffc0d0?, 0xc002533ef8)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:359 +0x5e
        github.com/ssbc/go-muxrpc/v2.(*rpc).serve(0xc0001324e0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:378 +0x19c
        github.com/ssbc/go-muxrpc/v2.Handle.func2()
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:128 +0x25
        created by github.com/ssbc/go-muxrpc/v2.Handle
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:127 +0x536
    leakcheck.go:113: Leaked goroutine: goroutine 77 [chan receive]:
        github.com/ssbc/go-muxrpc/v2.(*rpc).Serve(0xc002268cc0?)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:336 +0x2b
        github.com/ssbc/go-ssb/sbot.unixSockServer.serve({{0xffc450, 0xc0000fd950}, {0xff5d40, 0xc0021f3ea0}, {0xffb140, 0xc0000fdce0}, {0x0, 0x0, 0x0}, {0xc0000fdd10, ...}, ...}, ...)
        	/home/go-ssb/go-ssb/sbot/options.go:228 +0x4f9
        created by github.com/ssbc/go-ssb/sbot.unixSockServer.accept
        	/home/go-ssb/go-ssb/sbot/options.go:205 +0x51f
    leakcheck.go:113: Leaked goroutine: goroutine 78 [select]:
        context.propagateCancel.func1()
        	/usr/lib/go-1.18/src/context/context.go:279 +0x9e
        created by context.propagateCancel
        	/usr/lib/go-1.18/src/context/context.go:278 +0x1cb
    leakcheck.go:113: Leaked goroutine: goroutine 80 [IO wait]:
        internal/poll.runtime_pollWait(0x7fac09550d78, 0x72)
        	/usr/lib/go-1.18/src/runtime/netpoll.go:302 +0x89
        internal/poll.(*pollDesc).wait(0xc002258900?, 0xc00227e6e0?, 0x0)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:83 +0x32
        internal/poll.(*pollDesc).waitRead(...)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:88
        internal/poll.(*FD).Read(0xc002258900, {0xc00227e6e0, 0x9, 0x9})
        	/usr/lib/go-1.18/src/internal/poll/fd_unix.go:167 +0x25a
        net.(*netFD).Read(0xc002258900, {0xc00227e6e0?, 0x38?, 0x40f845?})
        	/usr/lib/go-1.18/src/net/fd_posix.go:55 +0x29
        net.(*conn).Read(0xc000130810, {0xc00227e6e0?, 0x7fac38b115b8?, 0x10?})
        	/usr/lib/go-1.18/src/net/net.go:183 +0x45
        io.ReadAtLeast({0x7fac0aed14b8, 0xc000141da0}, {0xc00227e6e0, 0x9, 0x9}, 0x9)
        	/usr/lib/go-1.18/src/io/io.go:331 +0x9a
        io.ReadFull(...)
        	/usr/lib/go-1.18/src/io/io.go:350
        encoding/binary.Read({0x7fac0aed14b8, 0xc000141da0}, {0xfff268, 0x15972e8}, {0xd3a9e0?, 0xc00227e6d0})
        	/usr/lib/go-1.18/src/encoding/binary/binary.go:256 +0xc3e
        github.com/ssbc/go-muxrpc/v2/codec.Reader.ReadHeader({{0x7fac0aed14b8?, 0xc000141da0?}}, 0xc00227e6d0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/codec/reader.go:48 +0x45
        github.com/ssbc/go-muxrpc/v2.(*Packer).NextHeader(0xc002268cc0, {0xffc0d0, 0xc00226a280}, 0xc00227e6d0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/packer.go:49 +0xcb
        github.com/ssbc/go-muxrpc/v2.(*rpc).serve.func2(0xc000132680, 0xffc0d0?, 0xc002531ef8)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:359 +0x5e
        github.com/ssbc/go-muxrpc/v2.(*rpc).serve(0xc000132680)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:378 +0x19c
        github.com/ssbc/go-muxrpc/v2.Handle.func2()
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:128 +0x25
        created by github.com/ssbc/go-muxrpc/v2.Handle
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:127 +0x536
    leakcheck.go:113: Leaked goroutine: goroutine 83 [chan receive]:
        github.com/ssbc/go-muxrpc/v2.(*rpc).Serve(0xc0001057d0?)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:336 +0x2b
        github.com/ssbc/go-ssb/client.NewUnix.func1()
        	/home/go-ssb/go-ssb/client/client.go:169 +0x4f
        created by github.com/ssbc/go-ssb/client.NewUnix
        	/home/go-ssb/go-ssb/client/client.go:168 +0x425
--- FAIL: TestNames (10.09s)
FAIL
```

So...it was the first one published that worked.  Maybe the `TODO: flush indexes` line has something to do with it?

After adding some logging statements in:

```
2022/12/10 22:13:08 [margaret/indexes/badger] overwrote batch limit 0
=== RUN   TestNames
2022/12/10 22:13:08 saved identity @8ozzaLJFihgAWLLCvI603RMcbDet9P0Hcf7UhI2edpQ=.ed25519 to testrun/TestNames/secrets/arny
2022/12/10 22:13:08 saved identity ssb:feed/gabbygrove-v1/KEjqhTG7P00RefFNeT8et1NdVIn0u7zEVps-RrBrf6M= to testrun/TestNames/secrets/bert
2022/12/10 22:13:08 saved identity @6wQOgjwJ5RwbxPD3j/tUuUG3xS8j0VxkpNMQwU9lyYs=.ed25519 to testrun/TestNames/secrets/cloe
2022/12/10 22:13:08 saved identity @ER5ndKPJWHzjJWSt+cBPABFJw4I+0LPfviVfukN8g84=.ed25519 to testrun/TestNames/secret
2022/12/10 22:13:08 created bot
2022/12/10 22:13:08 published messages
2022/12/10 22:13:08 TODO: flush indexes
2022/12/10 22:13:08 client joined socket
2022/12/10 22:13:08 got names
    names_test.go:128: 
        	Error Trace:	/home/go-ssb/go-ssb/sbot/names_test.go:128
        	Error:      	Not equal: 
        	            	expected: "i'm cloe!"
        	            	actual  : "that cloe"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-i'm cloe!
        	            	+that cloe
        	Test:       	TestNames
        	Messages:   	did not the right common name for cloe
level=debug event=update-replicate feed-want-count=0 hops=0 took=136.521µs
    leakcheck.go:113: Leaked goroutine: goroutine 10 [IO wait]:
        internal/poll.runtime_pollWait(0x7f3ac20e6658, 0x72)
        	/usr/lib/go-1.18/src/runtime/netpoll.go:302 +0x89
        internal/poll.(*pollDesc).wait(0xc00008f000?, 0x203000?, 0x0)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:83 +0x32
        internal/poll.(*pollDesc).waitRead(...)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:88
        internal/poll.(*FD).Accept(0xc00008f000)
        	/usr/lib/go-1.18/src/internal/poll/fd_unix.go:614 +0x22c
        net.(*netFD).accept(0xc00008f000)
        	/usr/lib/go-1.18/src/net/fd_unix.go:172 +0x35
        net.(*UnixListener).accept(0x4433c0?)
        	/usr/lib/go-1.18/src/net/unixsock_posix.go:166 +0x1c
        net.(*UnixListener).Accept(0xc002240f30)
        	/usr/lib/go-1.18/src/net/unixsock.go:260 +0x3d
        github.com/ssbc/go-ssb/sbot.unixSockServer.accept({{0xffc490, 0xc002240ba0}, {0xff5d80, 0xc002242cf0}, {0xffb180, 0xc002240f30}, {0x0, 0x0, 0x0}, {0xc002240f60, ...}, ...})
        	/home/go-ssb/go-ssb/sbot/options.go:175 +0x64
        created by github.com/ssbc/go-ssb/sbot.WithUNIXSocket.func1
        	/home/go-ssb/go-ssb/sbot/options.go:153 +0x33b
    leakcheck.go:113: Leaked goroutine: goroutine 11 [select]:
        github.com/ssbc/margaret/offset2.(*offsetQuery).Push(0xc00011a600, {0xffc490, 0xc002240ba0}, {0x7f3ac0034998?, 0xc000010298?})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/offset2/qry.go:202 +0xc9
        github.com/ssbc/go-luigi.Pump({0xffc490, 0xc002240ba0}, {0x7f3ac0034998?, 0xc000010298?}, {0xff5c80, 0xc00011a600})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:56 +0x82
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 12 [select]:
        github.com/dgraph-io/badger/v3/y.(*WaterMark).process(0xc002241770, 0xc002241740)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/y/watermark.go:214 +0x285
        created by github.com/dgraph-io/badger/v3/y.(*WaterMark).Init
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/y/watermark.go:72 +0xaa
    leakcheck.go:113: Leaked goroutine: goroutine 13 [select]:
        github.com/dgraph-io/badger/v3/y.(*WaterMark).process(0xc0022417a0, 0xc002241740)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/y/watermark.go:214 +0x285
        created by github.com/dgraph-io/badger/v3/y.(*WaterMark).Init
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/y/watermark.go:72 +0xaa
    leakcheck.go:113: Leaked goroutine: goroutine 14 [select]:
        github.com/dgraph-io/ristretto/z.(*AllocatorPool).freeupAllocators(0xc00000f3f8)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/z/allocator.go:385 +0x150
        created by github.com/dgraph-io/ristretto/z.NewAllocatorPool
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/z/allocator.go:324 +0xc5
    leakcheck.go:113: Leaked goroutine: goroutine 15 [select]:
        github.com/dgraph-io/ristretto.(*defaultPolicy).processItems(0xc000081a80)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/policy.go:102 +0x91
        created by github.com/dgraph-io/ristretto.newDefaultPolicy
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/policy.go:86 +0x156
    leakcheck.go:113: Leaked goroutine: goroutine 16 [select]:
        github.com/dgraph-io/ristretto.(*Cache).processItems(0xc00008f480)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/cache.go:452 +0x15e
        created by github.com/dgraph-io/ristretto.NewCache
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/cache.go:207 +0x696
    leakcheck.go:113: Leaked goroutine: goroutine 34 [select]:
        github.com/dgraph-io/badger/v3.(*DB).monitorCache(0xc0000b3200, 0xc002327a10)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:469 +0x18a
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:311 +0xc8b
    leakcheck.go:113: Leaked goroutine: goroutine 35 [select]:
        github.com/dgraph-io/badger/v3.(*DB).updateSize(0xc0000b3200, 0xc002327b60)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:1171 +0x158
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:331 +0xe8c
    leakcheck.go:113: Leaked goroutine: goroutine 36 [select]:
        github.com/dgraph-io/badger/v3.(*levelsController).runCompactor(0xc0001de000, 0x0, 0xc002327a40)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:508 +0x30d
        created by github.com/dgraph-io/badger/v3.(*levelsController).startCompact
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:354 +0x4e
    leakcheck.go:113: Leaked goroutine: goroutine 37 [select]:
        github.com/dgraph-io/badger/v3.(*levelsController).runCompactor(0xc0001de000, 0x1, 0xc002327a40)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:508 +0x30d
        created by github.com/dgraph-io/badger/v3.(*levelsController).startCompact
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:354 +0x4e
    leakcheck.go:113: Leaked goroutine: goroutine 38 [select]:
        github.com/dgraph-io/badger/v3.(*levelsController).runCompactor(0xc0001de000, 0x2, 0xc002327a40)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:508 +0x30d
        created by github.com/dgraph-io/badger/v3.(*levelsController).startCompact
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:354 +0x4e
    leakcheck.go:113: Leaked goroutine: goroutine 39 [select]:
        github.com/dgraph-io/badger/v3.(*levelsController).runCompactor(0xc0001de000, 0x3, 0xc002327a40)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:508 +0x30d
        created by github.com/dgraph-io/badger/v3.(*levelsController).startCompact
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:354 +0x4e
    leakcheck.go:113: Leaked goroutine: goroutine 40 [chan receive]:
        github.com/dgraph-io/badger/v3.(*DB).flushMemtable(0xc0000b3200, 0xc002241740?)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:1078 +0xb2
        github.com/dgraph-io/badger/v3.Open.func5()
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:357 +0x25
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:356 +0x107c
    leakcheck.go:113: Leaked goroutine: goroutine 41 [select]:
        github.com/dgraph-io/badger/v3.(*vlogThreshold).listenForValueThresholdUpdate(0xc000081a00)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/value.go:1172 +0x11a
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:380 +0x16dc
    leakcheck.go:113: Leaked goroutine: goroutine 42 [select]:
        github.com/dgraph-io/badger/v3.(*DB).doWrites(0xc0000b3200, 0xc002327bc0)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:900 +0x236
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:387 +0x179c
    leakcheck.go:113: Leaked goroutine: goroutine 43 [chan receive]:
        github.com/dgraph-io/badger/v3.(*valueLog).waitOnGC(0xc0000b33f8, 0x0?)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/value.go:1079 +0x7d
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:391 +0x185b
    leakcheck.go:113: Leaked goroutine: goroutine 44 [select]:
        github.com/dgraph-io/badger/v3.(*publisher).listenForUpdates(0xc002241860, 0xc002327c20)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/publisher.go:73 +0x150
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:395 +0x190a
    leakcheck.go:113: Leaked goroutine: goroutine 45 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc00008a0f0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 46 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc00008a230)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 47 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc00008a2d0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 48 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc00008a370)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 49 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc000186280)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 50 [select]:
        github.com/ssbc/margaret/offset2.(*offsetQuery).Push(0xc0001e3b00, {0xffc490, 0xc002240ba0}, {0x7f3ac0045e38?, 0xc00000e228?})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/offset2/qry.go:202 +0xc9
        github.com/ssbc/go-luigi.Pump({0xffc490, 0xc002240ba0}, {0x7f3ac0045e38?, 0xc00000e228?}, {0xff5c80, 0xc0001e3b00})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:56 +0x82
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 51 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc0001865a0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 52 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc00008a690)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 53 [select]:
        github.com/ssbc/margaret/offset2.(*offsetQuery).Push(0xc0000d2900, {0xffc490, 0xc002240ba0}, {0x7f3aa3225a98?, 0xc0001a0000?})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/offset2/qry.go:202 +0xc9
        github.com/ssbc/go-luigi.Pump({0xffc490, 0xc002240ba0}, {0x7f3aa3225a98?, 0xc0001a0000?}, {0xff5c80, 0xc0000d2900})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:56 +0x82
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 54 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc000186640)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 55 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*query).livequery(0xc002379e30, {0xffc490, 0xc002240ba0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:158 +0xe7
        github.com/ssbc/margaret/multilog/roaring.(*query).Next(0xc002379e30, {0xffc490, 0xc002240ba0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:126 +0x328
        github.com/ssbc/go-luigi/mfr.(*srcMap).Next(0xc0001256b0, {0xffc490, 0xc002240ba0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/mfr/map.go:58 +0x33
        github.com/ssbc/go-luigi.Pump({0xffc490, 0xc002240ba0}, {0x7f3ac0045e38?, 0xc00000eaf8?}, {0xff5760, 0xc0001256b0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:60 +0xac
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 56 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc000186780)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 57 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*query).livequery(0xc0001eae10, {0xffc490, 0xc002240ba0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:158 +0xe7
        github.com/ssbc/margaret/multilog/roaring.(*query).Next(0xc0001eae10, {0xffc490, 0xc002240ba0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:126 +0x328
        github.com/ssbc/go-luigi/mfr.(*srcMap).Next(0xc0001e47e0, {0xffc490, 0xc002240ba0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/mfr/map.go:58 +0x33
        github.com/ssbc/go-luigi.Pump({0xffc490, 0xc002240ba0}, {0x7f3ac0045e38?, 0xc00000eb88?}, {0xff5760, 0xc0001e47e0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:60 +0xac
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 58 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc000186960)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 59 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*query).livequery(0xc0001eacf0, {0xffc490, 0xc002240ba0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:158 +0xe7
        github.com/ssbc/margaret/multilog/roaring.(*query).Next(0xc0001eacf0, {0xffc490, 0xc002240ba0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:126 +0x328
        github.com/ssbc/go-luigi/mfr.(*srcMap).Next(0xc0001e4708, {0xffc490, 0xc002240ba0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/mfr/map.go:58 +0x33
        github.com/ssbc/go-luigi.Pump({0xffc490, 0xc002240ba0}, {0x7f3ac0045e38?, 0xc00000ec48?}, {0xff5760, 0xc0001e4708})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:60 +0xac
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 60 [select]:
        github.com/ssbc/go-ssb/sbot.debounce({0xffc490, 0xc002240ba0}, 0xb2d05e00, {0xffa070, 0xc000081740}, 0xc000032e40)
        	/home/go-ssb/go-ssb/sbot/replicate.go:135 +0x16c
        created by github.com/ssbc/go-ssb/sbot.(*Sbot).newGraphReplicator
        	/home/go-ssb/go-ssb/sbot/replicate.go:73 +0x3c5
    leakcheck.go:113: Leaked goroutine: goroutine 61 [select]:
        github.com/ssbc/margaret/offset2.(*offsetQuery).Push(0xc0000d2240, {0xffc490, 0xc002240ba0}, {0xff9430?, 0xc0000437b0?})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/offset2/qry.go:202 +0xc9
        github.com/ssbc/go-luigi.Pump({0xffc490, 0xc002240ba0}, {0xff9430?, 0xc0000437b0?}, {0xff5c80, 0xc0000d2240})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:56 +0x82
        github.com/ssbc/go-ssb/plugins/gossip.(*FeedManager).serveLiveFeeds(0xc00017e3f0)
        	/home/go-ssb/go-ssb/plugins/gossip/feed_manager.go:106 +0x168
        created by github.com/ssbc/go-ssb/plugins/gossip.NewFeedManager
        	/home/go-ssb/go-ssb/plugins/gossip/feed_manager.go:69 +0x1ea
    leakcheck.go:113: Leaked goroutine: goroutine 76 [IO wait]:
        internal/poll.runtime_pollWait(0x7f3ac20e6568, 0x72)
        	/usr/lib/go-1.18/src/runtime/netpoll.go:302 +0x89
        internal/poll.(*pollDesc).wait(0xc0001e8c00?, 0xc002244390?, 0x0)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:83 +0x32
        internal/poll.(*pollDesc).waitRead(...)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:88
        internal/poll.(*FD).Read(0xc0001e8c00, {0xc002244390, 0x9, 0x9})
        	/usr/lib/go-1.18/src/internal/poll/fd_unix.go:167 +0x25a
        net.(*netFD).Read(0xc0001e8c00, {0xc002244390?, 0x10000005423fc?, 0x7f3ac1081d20?})
        	/usr/lib/go-1.18/src/net/fd_posix.go:55 +0x29
        net.(*conn).Read(0xc000118128, {0xc002244390?, 0xc00222faf8?, 0x44fef2?})
        	/usr/lib/go-1.18/src/net/net.go:183 +0x45
        io.ReadAtLeast({0xff64e0, 0xc000118128}, {0xc002244390, 0x9, 0x9}, 0x9)
        	/usr/lib/go-1.18/src/io/io.go:331 +0x9a
        io.ReadFull(...)
        	/usr/lib/go-1.18/src/io/io.go:350
        encoding/binary.Read({0xff64e0, 0xc000118128}, {0xfff2a8, 0x15972e8}, {0xd3a9e0?, 0xc002244384})
        	/usr/lib/go-1.18/src/encoding/binary/binary.go:256 +0xc3e
        github.com/ssbc/go-muxrpc/v2/codec.Reader.ReadHeader({{0xff64e0?, 0xc000118128?}}, 0xc002244384)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/codec/reader.go:48 +0x45
        github.com/ssbc/go-muxrpc/v2.(*Packer).NextHeader(0xc0022220c0, {0xffc110, 0xc0001f6a40}, 0xc002244384)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/packer.go:49 +0xcb
        github.com/ssbc/go-muxrpc/v2.(*rpc).serve.func2(0xc0001cedd0, 0xffc110?, 0xc00222fef8)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:359 +0x5e
        github.com/ssbc/go-muxrpc/v2.(*rpc).serve(0xc0001cedd0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:378 +0x19c
        github.com/ssbc/go-muxrpc/v2.Handle.func2()
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:128 +0x25
        created by github.com/ssbc/go-muxrpc/v2.Handle
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:127 +0x536
    leakcheck.go:113: Leaked goroutine: goroutine 77 [chan receive]:
        github.com/ssbc/go-muxrpc/v2.(*rpc).Serve(0xc002222240?)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:336 +0x2b
        github.com/ssbc/go-ssb/sbot.unixSockServer.serve({{0xffc490, 0xc002240ba0}, {0xff5d80, 0xc002242cf0}, {0xffb180, 0xc002240f30}, {0x0, 0x0, 0x0}, {0xc002240f60, ...}, ...}, ...)
        	/home/go-ssb/go-ssb/sbot/options.go:228 +0x4f9
        created by github.com/ssbc/go-ssb/sbot.unixSockServer.accept
        	/home/go-ssb/go-ssb/sbot/options.go:205 +0x51f
    leakcheck.go:113: Leaked goroutine: goroutine 78 [select]:
        context.propagateCancel.func1()
        	/usr/lib/go-1.18/src/context/context.go:279 +0x9e
        created by context.propagateCancel
        	/usr/lib/go-1.18/src/context/context.go:278 +0x1cb
    leakcheck.go:113: Leaked goroutine: goroutine 8 [select]:
        github.com/ssbc/go-ssb/internal/ctxutils.WithError.func2()
        	/home/go-ssb/go-ssb/internal/ctxutils/withErr.go:31 +0x92
        created by github.com/ssbc/go-ssb/internal/ctxutils.WithError
        	/home/go-ssb/go-ssb/internal/ctxutils/withErr.go:30 +0x1df
    leakcheck.go:113: Leaked goroutine: goroutine 80 [IO wait]:
        internal/poll.runtime_pollWait(0x7f3ac20e6478, 0x72)
        	/usr/lib/go-1.18/src/runtime/netpoll.go:302 +0x89
        internal/poll.(*pollDesc).wait(0xc0001e9000?, 0xc002244200?, 0x0)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:83 +0x32
        internal/poll.(*pollDesc).waitRead(...)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:88
        internal/poll.(*FD).Read(0xc0001e9000, {0xc002244200, 0x9, 0x9})
        	/usr/lib/go-1.18/src/internal/poll/fd_unix.go:167 +0x25a
        net.(*netFD).Read(0xc0001e9000, {0xc002244200?, 0x38?, 0x40f845?})
        	/usr/lib/go-1.18/src/net/fd_posix.go:55 +0x29
        net.(*conn).Read(0xc000118378, {0xc002244200?, 0x7f3ae96b25b8?, 0x10?})
        	/usr/lib/go-1.18/src/net/net.go:183 +0x45
        io.ReadAtLeast({0x7f3ac0045f80, 0xc0001b6be0}, {0xc002244200, 0x9, 0x9}, 0x9)
        	/usr/lib/go-1.18/src/io/io.go:331 +0x9a
        io.ReadFull(...)
        	/usr/lib/go-1.18/src/io/io.go:350
        encoding/binary.Read({0x7f3ac0045f80, 0xc0001b6be0}, {0xfff2a8, 0x15972e8}, {0xd3a9e0?, 0xc0022441f0})
        	/usr/lib/go-1.18/src/encoding/binary/binary.go:256 +0xc3e
        github.com/ssbc/go-muxrpc/v2/codec.Reader.ReadHeader({{0x7f3ac0045f80?, 0xc0001b6be0?}}, 0xc0022441f0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/codec/reader.go:48 +0x45
        github.com/ssbc/go-muxrpc/v2.(*Packer).NextHeader(0xc002222240, {0xffc110, 0xc0001f6b40}, 0xc0022441f0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/packer.go:49 +0xcb
        github.com/ssbc/go-muxrpc/v2.(*rpc).serve.func2(0xc0001cef70, 0xffc110?, 0xc0001ffef8)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:359 +0x5e
        github.com/ssbc/go-muxrpc/v2.(*rpc).serve(0xc0001cef70)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:378 +0x19c
        github.com/ssbc/go-muxrpc/v2.Handle.func2()
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:128 +0x25
        created by github.com/ssbc/go-muxrpc/v2.Handle
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:127 +0x536
    leakcheck.go:113: Leaked goroutine: goroutine 9 [chan receive]:
        github.com/ssbc/go-ssb/blobstore.(*WantManager).replicateLoop(0xc0001863c0)
        	/home/go-ssb/go-ssb/blobstore/wants.go:94 +0x57
        created by github.com/ssbc/go-ssb/blobstore.NewWantManager
        	/home/go-ssb/go-ssb/blobstore/wants.go:60 +0x30a
    leakcheck.go:113: Leaked goroutine: goroutine 99 [chan receive]:
        github.com/ssbc/go-muxrpc/v2.(*rpc).Serve(0xc002344fd0?)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:336 +0x2b
        github.com/ssbc/go-ssb/client.NewUnix.func1()
        	/home/go-ssb/go-ssb/client/client.go:169 +0x4f
        created by github.com/ssbc/go-ssb/client.NewUnix
        	/home/go-ssb/go-ssb/client/client.go:168 +0x425
--- FAIL: TestNames (10.05s)
FAIL
```

Wherever the 10 second delay's happening, it appears it's after the test has failed out.

`WaitUntilIndexesAreSynced` didn't fix it:

```
2022/12/10 22:28:32 [margaret/indexes/badger] overwrote batch limit 0
=== RUN   TestNames
2022/12/10 22:28:32 saved identity @7YOLr65pmqedwFLvsKM9/ia8CR7Z2OtRiZSu5AAXTLc=.ed25519 to testrun/TestNames/secrets/arny
2022/12/10 22:28:32 saved identity ssb:feed/gabbygrove-v1/k03MOKWBMNO0Bng5lYnm69HT1gW6c-3mbWaHmkjKxjw= to testrun/TestNames/secrets/bert
2022/12/10 22:28:32 saved identity @TQHIiQY4kn+bQzLuwDBDPzVM+IPsdQHFoHva1edUBec=.ed25519 to testrun/TestNames/secrets/cloe
2022/12/10 22:28:32 saved identity @MMxUhgkePVVEF1yMbi8lwk1d/q9vEYizV14HTrofSG4=.ed25519 to testrun/TestNames/secret
2022/12/10 22:28:32 created bot
2022/12/10 22:28:32 published messages
2022/12/10 22:28:32 TODO: flush indexes
2022/12/10 22:28:32 client joined socket
2022/12/10 22:28:32 got names
    names_test.go:129: 
        	Error Trace:	/home/go-ssb/go-ssb/sbot/names_test.go:129
        	Error:      	Not equal: 
        	            	expected: "i'm cloe!"
        	            	actual  : "iditot"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-i'm cloe!
        	            	+iditot
        	Test:       	TestNames
        	Messages:   	did not the right common name for cloe
level=debug event=update-replicate feed-want-count=0 hops=0 took=128.157µs
    leakcheck.go:113: Leaked goroutine: goroutine 10 [IO wait]:
        internal/poll.runtime_pollWait(0x7fe6fc1ee458, 0x72)
        	/usr/lib/go-1.18/src/runtime/netpoll.go:302 +0x89
        internal/poll.(*pollDesc).wait(0xc0000ef100?, 0x203000?, 0x0)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:83 +0x32
        internal/poll.(*pollDesc).waitRead(...)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:88
        internal/poll.(*FD).Accept(0xc0000ef100)
        	/usr/lib/go-1.18/src/internal/poll/fd_unix.go:614 +0x22c
        net.(*netFD).accept(0xc0000ef100)
        	/usr/lib/go-1.18/src/net/fd_unix.go:172 +0x35
        net.(*UnixListener).accept(0x4433c0?)
        	/usr/lib/go-1.18/src/net/unixsock_posix.go:166 +0x1c
        net.(*UnixListener).Accept(0xc0021dfb30)
        	/usr/lib/go-1.18/src/net/unixsock.go:260 +0x3d
        github.com/ssbc/go-ssb/sbot.unixSockServer.accept({{0xffc490, 0xc0021df7a0}, {0xff5d80, 0xc0021c7e40}, {0xffb180, 0xc0021dfb30}, {0x0, 0x0, 0x0}, {0xc0021dfb60, ...}, ...})
        	/home/go-ssb/go-ssb/sbot/options.go:175 +0x64
        created by github.com/ssbc/go-ssb/sbot.WithUNIXSocket.func1
        	/home/go-ssb/go-ssb/sbot/options.go:153 +0x33b
    leakcheck.go:113: Leaked goroutine: goroutine 101 [chan receive]:
        github.com/ssbc/go-muxrpc/v2.(*rpc).Serve(0xc0078cffd0?)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:336 +0x2b
        github.com/ssbc/go-ssb/client.NewUnix.func1()
        	/home/go-ssb/go-ssb/client/client.go:169 +0x4f
        created by github.com/ssbc/go-ssb/client.NewUnix
        	/home/go-ssb/go-ssb/client/client.go:168 +0x425
    leakcheck.go:113: Leaked goroutine: goroutine 11 [select]:
        github.com/ssbc/margaret/offset2.(*offsetQuery).Push(0xc000032600, {0xffc490, 0xc0021df7a0}, {0x7fe6fca32218?, 0xc0000cc728?})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/offset2/qry.go:202 +0xc9
        github.com/ssbc/go-luigi.Pump({0xffc490, 0xc0021df7a0}, {0x7fe6fca32218?, 0xc0000cc728?}, {0xff5c80, 0xc000032600})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:56 +0x82
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 12 [select]:
        github.com/dgraph-io/badger/v3/y.(*WaterMark).process(0xc0022ce390, 0xc0022ce360)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/y/watermark.go:214 +0x285
        created by github.com/dgraph-io/badger/v3/y.(*WaterMark).Init
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/y/watermark.go:72 +0xaa
    leakcheck.go:113: Leaked goroutine: goroutine 13 [select]:
        github.com/dgraph-io/badger/v3/y.(*WaterMark).process(0xc0022ce3c0, 0xc0022ce360)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/y/watermark.go:214 +0x285
        created by github.com/dgraph-io/badger/v3/y.(*WaterMark).Init
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/y/watermark.go:72 +0xaa
    leakcheck.go:113: Leaked goroutine: goroutine 14 [select]:
        github.com/dgraph-io/ristretto/z.(*AllocatorPool).freeupAllocators(0xc0000c9860)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/z/allocator.go:385 +0x150
        created by github.com/dgraph-io/ristretto/z.NewAllocatorPool
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/z/allocator.go:324 +0xc5
    leakcheck.go:113: Leaked goroutine: goroutine 15 [select]:
        github.com/dgraph-io/ristretto.(*defaultPolicy).processItems(0xc000133c00)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/policy.go:102 +0x91
        created by github.com/dgraph-io/ristretto.newDefaultPolicy
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/policy.go:86 +0x156
    leakcheck.go:113: Leaked goroutine: goroutine 16 [select]:
        github.com/dgraph-io/ristretto.(*Cache).processItems(0xc0000ef580)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/cache.go:452 +0x15e
        created by github.com/dgraph-io/ristretto.NewCache
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/cache.go:207 +0x696
    leakcheck.go:113: Leaked goroutine: goroutine 24 [select]:
        github.com/dgraph-io/badger/v3.(*levelsController).runCompactor(0xc00227a070, 0x0, 0xc0022e8810)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:508 +0x30d
        created by github.com/dgraph-io/badger/v3.(*levelsController).startCompact
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:354 +0x4e
    leakcheck.go:113: Leaked goroutine: goroutine 25 [select]:
        github.com/dgraph-io/badger/v3.(*levelsController).runCompactor(0xc00227a070, 0x1, 0xc0022e8810)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:508 +0x30d
        created by github.com/dgraph-io/badger/v3.(*levelsController).startCompact
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:354 +0x4e
    leakcheck.go:113: Leaked goroutine: goroutine 26 [select]:
        github.com/dgraph-io/badger/v3.(*levelsController).runCompactor(0xc00227a070, 0x2, 0xc0022e8810)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:508 +0x30d
        created by github.com/dgraph-io/badger/v3.(*levelsController).startCompact
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:354 +0x4e
    leakcheck.go:113: Leaked goroutine: goroutine 27 [select]:
        github.com/dgraph-io/badger/v3.(*levelsController).runCompactor(0xc00227a070, 0x3, 0xc0022e8810)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:508 +0x30d
        created by github.com/dgraph-io/badger/v3.(*levelsController).startCompact
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:354 +0x4e
    leakcheck.go:113: Leaked goroutine: goroutine 28 [chan receive]:
        github.com/dgraph-io/badger/v3.(*DB).flushMemtable(0xc0000d5200, 0x0?)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:1078 +0xb2
        github.com/dgraph-io/badger/v3.Open.func5()
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:357 +0x25
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:356 +0x107c
    leakcheck.go:113: Leaked goroutine: goroutine 29 [select]:
        github.com/dgraph-io/badger/v3.(*vlogThreshold).listenForValueThresholdUpdate(0xc000133b80)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/value.go:1172 +0x11a
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:380 +0x16dc
    leakcheck.go:113: Leaked goroutine: goroutine 30 [select]:
        github.com/dgraph-io/badger/v3.(*DB).doWrites(0xc0000d5200, 0xc0022e88d0)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:900 +0x236
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:387 +0x179c
    leakcheck.go:113: Leaked goroutine: goroutine 31 [chan receive]:
        github.com/dgraph-io/badger/v3.(*valueLog).waitOnGC(0xc0000d53f8, 0xc0022ce360?)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/value.go:1079 +0x7d
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:391 +0x185b
    leakcheck.go:113: Leaked goroutine: goroutine 32 [select]:
        github.com/dgraph-io/badger/v3.(*publisher).listenForUpdates(0xc0022ce480, 0xc0022e8930)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/publisher.go:73 +0x150
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:395 +0x190a
    leakcheck.go:113: Leaked goroutine: goroutine 33 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc00011a1e0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 34 [select]:
        github.com/dgraph-io/badger/v3.(*DB).monitorCache(0xc0000d5200, 0xc0022e8630)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:469 +0x18a
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:311 +0xc8b
    leakcheck.go:113: Leaked goroutine: goroutine 35 [select]:
        github.com/dgraph-io/badger/v3.(*DB).updateSize(0xc0000d5200, 0xc0022e8780)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:1171 +0x158
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:331 +0xe8c
    leakcheck.go:113: Leaked goroutine: goroutine 50 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc00011a2d0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 51 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc00011a370)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 52 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc00011a410)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 53 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc0000e46e0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 54 [select]:
        github.com/ssbc/margaret/offset2.(*offsetQuery).Push(0xc000033920, {0xffc490, 0xc0021df7a0}, {0x7fe6fc17fa78?, 0xc0000c85e8?})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/offset2/qry.go:202 +0xc9
        github.com/ssbc/go-luigi.Pump({0xffc490, 0xc0021df7a0}, {0x7fe6fc17fa78?, 0xc0000c85e8?}, {0xff5c80, 0xc000033920})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:56 +0x82
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 55 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc0000e4780)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 56 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc00011a7d0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 57 [select]:
        github.com/ssbc/margaret/offset2.(*offsetQuery).Push(0xc000033ce0, {0xffc490, 0xc0021df7a0}, {0x7fe6fc17fb20?, 0xc0000c0000?})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/offset2/qry.go:202 +0xc9
        github.com/ssbc/go-luigi.Pump({0xffc490, 0xc0021df7a0}, {0x7fe6fc17fb20?, 0xc0000c0000?}, {0xff5c80, 0xc000033ce0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:56 +0x82
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 58 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc0000e4820)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 59 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*query).livequery(0xc00007f3e0, {0xffc490, 0xc0021df7a0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:158 +0xe7
        github.com/ssbc/margaret/multilog/roaring.(*query).Next(0xc00007f3e0, {0xffc490, 0xc0021df7a0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:126 +0x328
        github.com/ssbc/go-luigi/mfr.(*srcMap).Next(0xc00000e4e0, {0xffc490, 0xc0021df7a0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/mfr/map.go:58 +0x33
        github.com/ssbc/go-luigi.Pump({0xffc490, 0xc0021df7a0}, {0x7fe6fc17fa78?, 0xc0000c86c0?}, {0xff5760, 0xc00000e4e0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:60 +0xac
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 60 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc0000e4960)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 61 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*query).livequery(0xc00007f500, {0xffc490, 0xc0021df7a0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:158 +0xe7
        github.com/ssbc/margaret/multilog/roaring.(*query).Next(0xc00007f500, {0xffc490, 0xc0021df7a0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:126 +0x328
        github.com/ssbc/go-luigi/mfr.(*srcMap).Next(0xc00000e5b8, {0xffc490, 0xc0021df7a0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/mfr/map.go:58 +0x33
        github.com/ssbc/go-luigi.Pump({0xffc490, 0xc0021df7a0}, {0x7fe6fc17fa78?, 0xc0000c8eb8?}, {0xff5760, 0xc00000e5b8})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:60 +0xac
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 62 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc0000e4b40)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 63 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*query).livequery(0xc00007f620, {0xffc490, 0xc0021df7a0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:158 +0xe7
        github.com/ssbc/margaret/multilog/roaring.(*query).Next(0xc00007f620, {0xffc490, 0xc0021df7a0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:126 +0x328
        github.com/ssbc/go-luigi/mfr.(*srcMap).Next(0xc00000e690, {0xffc490, 0xc0021df7a0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/mfr/map.go:58 +0x33
        github.com/ssbc/go-luigi.Pump({0xffc490, 0xc0021df7a0}, {0x7fe6fc17fa78?, 0xc0000c9008?}, {0xff5760, 0xc00000e690})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:60 +0xac
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:202 +0xa05
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 64 [select]:
        github.com/ssbc/go-ssb/sbot.debounce({0xffc490, 0xc0021df7a0}, 0xb2d05e00, {0xffa070, 0xc0001338c0}, 0xc000091380)
        	/home/go-ssb/go-ssb/sbot/replicate.go:135 +0x16c
        created by github.com/ssbc/go-ssb/sbot.(*Sbot).newGraphReplicator
        	/home/go-ssb/go-ssb/sbot/replicate.go:73 +0x3c5
    leakcheck.go:113: Leaked goroutine: goroutine 65 [select]:
        github.com/ssbc/margaret/offset2.(*offsetQuery).Push(0xc0022fe6c0, {0xffc490, 0xc0021df7a0}, {0xff9430?, 0xc000043130?})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/offset2/qry.go:202 +0xc9
        github.com/ssbc/go-luigi.Pump({0xffc490, 0xc0021df7a0}, {0xff9430?, 0xc000043130?}, {0xff5c80, 0xc0022fe6c0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:56 +0x82
        github.com/ssbc/go-ssb/plugins/gossip.(*FeedManager).serveLiveFeeds(0xc00227a460)
        	/home/go-ssb/go-ssb/plugins/gossip/feed_manager.go:106 +0x168
        created by github.com/ssbc/go-ssb/plugins/gossip.NewFeedManager
        	/home/go-ssb/go-ssb/plugins/gossip/feed_manager.go:69 +0x1ea
    leakcheck.go:113: Leaked goroutine: goroutine 78 [IO wait]:
        internal/poll.runtime_pollWait(0x7fe6fc1ee368, 0x72)
        	/usr/lib/go-1.18/src/runtime/netpoll.go:302 +0x89
        internal/poll.(*pollDesc).wait(0xc0022fb480?, 0xc00790cbd0?, 0x0)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:83 +0x32
        internal/poll.(*pollDesc).waitRead(...)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:88
        internal/poll.(*FD).Read(0xc0022fb480, {0xc00790cbd0, 0x9, 0x9})
        	/usr/lib/go-1.18/src/internal/poll/fd_unix.go:167 +0x25a
        net.(*netFD).Read(0xc0022fb480, {0xc00790cbd0?, 0x10000005423fc?, 0x7fe6fc20a878?})
        	/usr/lib/go-1.18/src/net/fd_posix.go:55 +0x29
        net.(*conn).Read(0xc000010100, {0xc00790cbd0?, 0xc002181af8?, 0x44fef2?})
        	/usr/lib/go-1.18/src/net/net.go:183 +0x45
        io.ReadAtLeast({0xff64e0, 0xc000010100}, {0xc00790cbd0, 0x9, 0x9}, 0x9)
        	/usr/lib/go-1.18/src/io/io.go:331 +0x9a
        io.ReadFull(...)
        	/usr/lib/go-1.18/src/io/io.go:350
        encoding/binary.Read({0xff64e0, 0xc000010100}, {0xfff2a8, 0x15972e8}, {0xd3a9e0?, 0xc00790cbc0})
        	/usr/lib/go-1.18/src/encoding/binary/binary.go:256 +0xc3e
        github.com/ssbc/go-muxrpc/v2/codec.Reader.ReadHeader({{0xff64e0?, 0xc000010100?}}, 0xc00790cbc0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/codec/reader.go:48 +0x45
        github.com/ssbc/go-muxrpc/v2.(*Packer).NextHeader(0xc0022ffc80, {0xffc110, 0xc0021dd640}, 0xc00790cbc0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/packer.go:49 +0xcb
        github.com/ssbc/go-muxrpc/v2.(*rpc).serve.func2(0xc000101e10, 0xffc110?, 0xc002181ef8)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:359 +0x5e
        github.com/ssbc/go-muxrpc/v2.(*rpc).serve(0xc000101e10)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:378 +0x19c
        github.com/ssbc/go-muxrpc/v2.Handle.func2()
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:128 +0x25
        created by github.com/ssbc/go-muxrpc/v2.Handle
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:127 +0x536
    leakcheck.go:113: Leaked goroutine: goroutine 79 [chan receive]:
        github.com/ssbc/go-muxrpc/v2.(*rpc).Serve(0xc0022fff80?)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:336 +0x2b
        github.com/ssbc/go-ssb/sbot.unixSockServer.serve({{0xffc490, 0xc0021df7a0}, {0xff5d80, 0xc0021c7e40}, {0xffb180, 0xc0021dfb30}, {0x0, 0x0, 0x0}, {0xc0021dfb60, ...}, ...}, ...)
        	/home/go-ssb/go-ssb/sbot/options.go:228 +0x4f9
        created by github.com/ssbc/go-ssb/sbot.unixSockServer.accept
        	/home/go-ssb/go-ssb/sbot/options.go:205 +0x51f
    leakcheck.go:113: Leaked goroutine: goroutine 8 [select]:
        github.com/ssbc/go-ssb/internal/ctxutils.WithError.func2()
        	/home/go-ssb/go-ssb/internal/ctxutils/withErr.go:31 +0x92
        created by github.com/ssbc/go-ssb/internal/ctxutils.WithError
        	/home/go-ssb/go-ssb/internal/ctxutils/withErr.go:30 +0x1df
    leakcheck.go:113: Leaked goroutine: goroutine 80 [select]:
        context.propagateCancel.func1()
        	/usr/lib/go-1.18/src/context/context.go:279 +0x9e
        created by context.propagateCancel
        	/usr/lib/go-1.18/src/context/context.go:278 +0x1cb
    leakcheck.go:113: Leaked goroutine: goroutine 9 [chan receive]:
        github.com/ssbc/go-ssb/blobstore.(*WantManager).replicateLoop(0xc000113f40)
        	/home/go-ssb/go-ssb/blobstore/wants.go:94 +0x57
        created by github.com/ssbc/go-ssb/blobstore.NewWantManager
        	/home/go-ssb/go-ssb/blobstore/wants.go:60 +0x30a
    leakcheck.go:113: Leaked goroutine: goroutine 98 [IO wait]:
        internal/poll.runtime_pollWait(0x7fe6fc1ee278, 0x72)
        	/usr/lib/go-1.18/src/runtime/netpoll.go:302 +0x89
        internal/poll.(*pollDesc).wait(0xc0022fb880?, 0xc00790ca80?, 0x0)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:83 +0x32
        internal/poll.(*pollDesc).waitRead(...)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:88
        internal/poll.(*FD).Read(0xc0022fb880, {0xc00790ca80, 0x9, 0x9})
        	/usr/lib/go-1.18/src/internal/poll/fd_unix.go:167 +0x25a
        net.(*netFD).Read(0xc0022fb880, {0xc00790ca80?, 0x38?, 0x40f845?})
        	/usr/lib/go-1.18/src/net/fd_posix.go:55 +0x29
        net.(*conn).Read(0xc000010118, {0xc00790ca80?, 0x7fe725059108?, 0x10?})
        	/usr/lib/go-1.18/src/net/net.go:183 +0x45
        io.ReadAtLeast({0x7fe6fc006998, 0xc0000751e0}, {0xc00790ca80, 0x9, 0x9}, 0x9)
        	/usr/lib/go-1.18/src/io/io.go:331 +0x9a
        io.ReadFull(...)
        	/usr/lib/go-1.18/src/io/io.go:350
        encoding/binary.Read({0x7fe6fc006998, 0xc0000751e0}, {0xfff2a8, 0x15972e8}, {0xd3a9e0?, 0xc00790ca70})
        	/usr/lib/go-1.18/src/encoding/binary/binary.go:256 +0xc3e
        github.com/ssbc/go-muxrpc/v2/codec.Reader.ReadHeader({{0x7fe6fc006998?, 0xc0000751e0?}}, 0xc00790ca70)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/codec/reader.go:48 +0x45
        github.com/ssbc/go-muxrpc/v2.(*Packer).NextHeader(0xc0022fff80, {0xffc110, 0xc0021dd740}, 0xc00790ca70)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/packer.go:49 +0xcb
        github.com/ssbc/go-muxrpc/v2.(*rpc).serve.func2(0xc0022a4000, 0xffc110?, 0xc002187ef8)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:359 +0x5e
        github.com/ssbc/go-muxrpc/v2.(*rpc).serve(0xc0022a4000)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:378 +0x19c
        github.com/ssbc/go-muxrpc/v2.Handle.func2()
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:128 +0x25
        created by github.com/ssbc/go-muxrpc/v2.Handle
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:127 +0x536
--- FAIL: TestNames (10.09s)
FAIL
```

`WaitUntilIndexesAreSynced` did seem to make a difference.  The failures are now consistently wrong rather than not having all the messages:

```
2022/12/10 23:18:33 [margaret/indexes/badger] overwrote batch limit 0
=== RUN   TestNames
2022/12/10 23:18:33 saved identity @RuQuXm8h4PR7q0laeFjFAJJzPY7IgzFbeml+uzRGjwc=.ed25519 to testrun/TestNames/secrets/arny
2022/12/10 23:18:33 saved identity ssb:feed/gabbygrove-v1/pPqPDTasasq0_P9Ea04wlNMeeydpc0eanMQIx6ln814= to testrun/TestNames/secrets/bert
2022/12/10 23:18:33 saved identity @VyR7ZLwpoKKS55O4KrQgP2nObbKPfktdaD1CY0SQBWI=.ed25519 to testrun/TestNames/secrets/cloe
2022/12/10 23:18:33 saved identity @fCpnmxPRkQAo25+47OWu3J4PCOfTTpF1mzCJdm0YKr0=.ed25519 to testrun/TestNames/secret
2022/12/10 23:18:33 created bot
2022/12/10 23:18:33 published messages
2022/12/10 23:18:33 TODO: flush indexes
2022/12/10 23:18:33 index states: map[string]string{"abouts":"live", "combined":"live", "contacts":"live", "get":"live", "group-members":"live", "timestamps":"live"}
2022/12/10 23:18:33 client joined socket
2022/12/10 23:18:33 got names
    names_test.go:130: 
        	Error Trace:	/home/go-ssb/go-ssb/sbot/names_test.go:130
        	Error:      	Not equal: 
        	            	expected: "i'm cloe!"
        	            	actual  : "i'm arny!"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-i'm cloe!
        	            	+i'm arny!
        	Test:       	TestNames
        	Messages:   	did not the right common name for cloe
level=debug event=update-replicate feed-want-count=0 hops=0 took=123.195µs
    leakcheck.go:113: Leaked goroutine: goroutine 10 [select]:
        github.com/ssbc/margaret/offset2.(*offsetQuery).Push(0xc0000323c0, {0xffc4d0, 0xc0022e25a0}, {0x7f9df52e9658?, 0xc00011a758?})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/offset2/qry.go:202 +0xc9
        github.com/ssbc/go-luigi.Pump({0xffc4d0, 0xc0022e25a0}, {0x7f9df52e9658?, 0xc00011a758?}, {0xff5cc0, 0xc0000323c0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:56 +0x82
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:204 +0xa07
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 11 [select]:
        github.com/dgraph-io/badger/v3/y.(*WaterMark).process(0xc0022e3170, 0xc0022e3140)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/y/watermark.go:214 +0x285
        created by github.com/dgraph-io/badger/v3/y.(*WaterMark).Init
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/y/watermark.go:72 +0xaa
    leakcheck.go:113: Leaked goroutine: goroutine 12 [select]:
        github.com/dgraph-io/badger/v3.(*DB).updateSize(0xc0001da000, 0xc0022e3290)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:1171 +0x158
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:331 +0xe8c
    leakcheck.go:113: Leaked goroutine: goroutine 25 [select]:
        github.com/dgraph-io/badger/v3/y.(*WaterMark).process(0xc0022e31a0, 0xc0022e3140)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/y/watermark.go:214 +0x285
        created by github.com/dgraph-io/badger/v3/y.(*WaterMark).Init
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/y/watermark.go:72 +0xaa
    leakcheck.go:113: Leaked goroutine: goroutine 26 [select]:
        github.com/dgraph-io/ristretto/z.(*AllocatorPool).freeupAllocators(0xc00000e0a8)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/z/allocator.go:385 +0x150
        created by github.com/dgraph-io/ristretto/z.NewAllocatorPool
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/z/allocator.go:324 +0xc5
    leakcheck.go:113: Leaked goroutine: goroutine 27 [select]:
        github.com/dgraph-io/ristretto.(*defaultPolicy).processItems(0xc000159780)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/policy.go:102 +0x91
        created by github.com/dgraph-io/ristretto.newDefaultPolicy
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/policy.go:86 +0x156
    leakcheck.go:113: Leaked goroutine: goroutine 28 [select]:
        github.com/dgraph-io/ristretto.(*Cache).processItems(0xc0001c4080)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/cache.go:452 +0x15e
        created by github.com/dgraph-io/ristretto.NewCache
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/ristretto@v0.1.1/cache.go:207 +0x696
    leakcheck.go:113: Leaked goroutine: goroutine 29 [select]:
        github.com/dgraph-io/badger/v3.(*DB).monitorCache(0xc0001da000, 0xc0001e4630)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:469 +0x18a
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:311 +0xc8b
    leakcheck.go:113: Leaked goroutine: goroutine 30 [select]:
        github.com/dgraph-io/badger/v3.(*levelsController).runCompactor(0xc000187490, 0x0, 0xc0022e2000)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:508 +0x30d
        created by github.com/dgraph-io/badger/v3.(*levelsController).startCompact
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:354 +0x4e
    leakcheck.go:113: Leaked goroutine: goroutine 31 [select]:
        github.com/dgraph-io/badger/v3.(*levelsController).runCompactor(0xc000187490, 0x1, 0xc0022e2000)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:508 +0x30d
        created by github.com/dgraph-io/badger/v3.(*levelsController).startCompact
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:354 +0x4e
    leakcheck.go:113: Leaked goroutine: goroutine 32 [select]:
        github.com/dgraph-io/badger/v3.(*levelsController).runCompactor(0xc000187490, 0x2, 0xc0022e2000)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:508 +0x30d
        created by github.com/dgraph-io/badger/v3.(*levelsController).startCompact
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:354 +0x4e
    leakcheck.go:113: Leaked goroutine: goroutine 33 [select]:
        github.com/dgraph-io/badger/v3.(*levelsController).runCompactor(0xc000187490, 0x3, 0xc0022e2000)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:508 +0x30d
        created by github.com/dgraph-io/badger/v3.(*levelsController).startCompact
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/levels.go:354 +0x4e
    leakcheck.go:113: Leaked goroutine: goroutine 34 [chan receive]:
        github.com/dgraph-io/badger/v3.(*DB).flushMemtable(0xc0001da000, 0x0?)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:1078 +0xb2
        github.com/dgraph-io/badger/v3.Open.func5()
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:357 +0x25
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:356 +0x107c
    leakcheck.go:113: Leaked goroutine: goroutine 35 [select]:
        github.com/dgraph-io/badger/v3.(*vlogThreshold).listenForValueThresholdUpdate(0xc000159700)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/value.go:1172 +0x11a
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:380 +0x16dc
    leakcheck.go:113: Leaked goroutine: goroutine 36 [select]:
        github.com/dgraph-io/badger/v3.(*DB).doWrites(0xc0001da000, 0xc0022e2150)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:900 +0x236
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:387 +0x179c
    leakcheck.go:113: Leaked goroutine: goroutine 37 [chan receive]:
        github.com/dgraph-io/badger/v3.(*valueLog).waitOnGC(0xc0001da1f8, 0x0?)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/value.go:1079 +0x7d
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:391 +0x185b
    leakcheck.go:113: Leaked goroutine: goroutine 38 [select]:
        github.com/dgraph-io/badger/v3.(*publisher).listenForUpdates(0xc000094480, 0xc0022e21b0)
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/publisher.go:73 +0x150
        created by github.com/dgraph-io/badger/v3.Open
        	/home/go-ssb/go/pkg/mod/github.com/dgraph-io/badger/v3@v3.2103.4/db.go:395 +0x190a
    leakcheck.go:113: Leaked goroutine: goroutine 39 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc00013e140)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 40 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc00013e2d0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 41 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc00013e370)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 42 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc00013e410)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 43 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc000188140)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 44 [select]:
        github.com/ssbc/margaret/offset2.(*offsetQuery).Push(0xc00223e660, {0xffc4d0, 0xc0022e25a0}, {0x7f9df44e01f8?, 0xc00012c6c0?})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/offset2/qry.go:202 +0xc9
        github.com/ssbc/go-luigi.Pump({0xffc4d0, 0xc0022e25a0}, {0x7f9df44e01f8?, 0xc00012c6c0?}, {0xff5cc0, 0xc00223e660})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:56 +0x82
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:204 +0xa07
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 45 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc000188320)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 46 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*MultiLog).writeBatches(0xc00013e730)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:46 +0x8e
        created by github.com/ssbc/margaret/multilog/roaring.NewStore
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/multilog.go:40 +0x1b6
    leakcheck.go:113: Leaked goroutine: goroutine 47 [select]:
        github.com/ssbc/margaret/offset2.(*offsetQuery).Push(0xc000101aa0, {0xffc4d0, 0xc0022e25a0}, {0x7f9df44e32f8?, 0xc0001a0000?})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/offset2/qry.go:202 +0xc9
        github.com/ssbc/go-luigi.Pump({0xffc4d0, 0xc0022e25a0}, {0x7f9df44e32f8?, 0xc0001a0000?}, {0xff5cc0, 0xc000101aa0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:56 +0x82
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:204 +0xa07
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 48 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc000188640)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 49 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*query).livequery(0xc00233e240, {0xffc4d0, 0xc0022e25a0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:158 +0xe7
        github.com/ssbc/margaret/multilog/roaring.(*query).Next(0xc00233e240, {0xffc4d0, 0xc0022e25a0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:126 +0x328
        github.com/ssbc/go-luigi/mfr.(*srcMap).Next(0xc00233a198, {0xffc4d0, 0xc0022e25a0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/mfr/map.go:58 +0x33
        github.com/ssbc/go-luigi.Pump({0xffc4d0, 0xc0022e25a0}, {0x7f9df44e01f8?, 0xc00012c858?}, {0xff57a0, 0xc00233a198})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:60 +0xac
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:204 +0xa07
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 50 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc000188780)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 51 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*query).livequery(0xc0022680c0, {0xffc4d0, 0xc0022e25a0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:158 +0xe7
        github.com/ssbc/margaret/multilog/roaring.(*query).Next(0xc0022680c0, {0xffc4d0, 0xc0022e25a0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:126 +0x328
        github.com/ssbc/go-luigi/mfr.(*srcMap).Next(0xc00000f8c0, {0xffc4d0, 0xc0022e25a0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/mfr/map.go:58 +0x33
        github.com/ssbc/go-luigi.Pump({0xffc4d0, 0xc0022e25a0}, {0x7f9df44e01f8?, 0xc00012c8e8?}, {0xff57a0, 0xc00000f8c0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:60 +0xac
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:204 +0xa07
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 52 [select]:
        github.com/ssbc/margaret/indexes/badger.(*index).writeBatches(0xc000188960)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:178 +0xa7
        created by github.com/ssbc/margaret/indexes/badger.newIndex
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/indexes/badger/index.go:102 +0x2a5
    leakcheck.go:113: Leaked goroutine: goroutine 53 [select]:
        github.com/ssbc/margaret/multilog/roaring.(*query).livequery(0xc0000a3f50, {0xffc4d0, 0xc0022e25a0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:158 +0xe7
        github.com/ssbc/margaret/multilog/roaring.(*query).Next(0xc0000a3f50, {0xffc4d0, 0xc0022e25a0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/multilog/roaring/qry.go:126 +0x328
        github.com/ssbc/go-luigi/mfr.(*srcMap).Next(0xc00000f7d0, {0xffc4d0, 0xc0022e25a0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/mfr/map.go:58 +0x33
        github.com/ssbc/go-luigi.Pump({0xffc4d0, 0xc0022e25a0}, {0x7f9df44e01f8?, 0xc00012d1a0?}, {0xff57a0, 0xc00000f7d0})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:60 +0xac
        github.com/ssbc/go-ssb/sbot.(*Sbot).serveIndexFrom.func1()
        	/home/go-ssb/go-ssb/sbot/indexes.go:204 +0xa07
        golang.org/x/sync/errgroup.(*Group).Go.func1()
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:75 +0x64
        created by golang.org/x/sync/errgroup.(*Group).Go
        	/home/go-ssb/go/pkg/mod/golang.org/x/sync@v0.1.0/errgroup/errgroup.go:72 +0xa5
    leakcheck.go:113: Leaked goroutine: goroutine 54 [select]:
        github.com/ssbc/go-ssb/sbot.debounce({0xffc4d0, 0xc0022e25a0}, 0xb2d05e00, {0xffa0b0, 0xc000081700}, 0xc002322ae0)
        	/home/go-ssb/go-ssb/sbot/replicate.go:135 +0x16c
        created by github.com/ssbc/go-ssb/sbot.(*Sbot).newGraphReplicator
        	/home/go-ssb/go-ssb/sbot/replicate.go:73 +0x3c5
    leakcheck.go:113: Leaked goroutine: goroutine 55 [select]:
        github.com/ssbc/margaret/offset2.(*offsetQuery).Push(0xc002323e60, {0xffc4d0, 0xc0022e25a0}, {0xff9470?, 0xc0000437b0?})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/margaret@v0.4.4-0.20221101112304-4f5815095ef3/offset2/qry.go:202 +0xc9
        github.com/ssbc/go-luigi.Pump({0xffc4d0, 0xc0022e25a0}, {0xff9470?, 0xc0000437b0?}, {0xff5cc0, 0xc002323e60})
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-luigi@v0.3.7-0.20221019204020-324065b9a7c6/stream.go:56 +0x82
        github.com/ssbc/go-ssb/plugins/gossip.(*FeedManager).serveLiveFeeds(0xc0001863f0)
        	/home/go-ssb/go-ssb/plugins/gossip/feed_manager.go:106 +0x168
        created by github.com/ssbc/go-ssb/plugins/gossip.NewFeedManager
        	/home/go-ssb/go-ssb/plugins/gossip/feed_manager.go:69 +0x1ea
    leakcheck.go:113: Leaked goroutine: goroutine 7 [select]:
        github.com/ssbc/go-ssb/internal/ctxutils.WithError.func2()
        	/home/go-ssb/go-ssb/internal/ctxutils/withErr.go:31 +0x92
        created by github.com/ssbc/go-ssb/internal/ctxutils.WithError
        	/home/go-ssb/go-ssb/internal/ctxutils/withErr.go:30 +0x1df
    leakcheck.go:113: Leaked goroutine: goroutine 8 [chan receive]:
        github.com/ssbc/go-ssb/blobstore.(*WantManager).replicateLoop(0xc000188460)
        	/home/go-ssb/go-ssb/blobstore/wants.go:94 +0x57
        created by github.com/ssbc/go-ssb/blobstore.NewWantManager
        	/home/go-ssb/go-ssb/blobstore/wants.go:60 +0x30a
    leakcheck.go:113: Leaked goroutine: goroutine 87 [IO wait]:
        internal/poll.runtime_pollWait(0x7f9deffcf8a8, 0x72)
        	/usr/lib/go-1.18/src/runtime/netpoll.go:302 +0x89
        internal/poll.(*pollDesc).wait(0xc0001c4880?, 0xc002324eb0?, 0x0)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:83 +0x32
        internal/poll.(*pollDesc).waitRead(...)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:88
        internal/poll.(*FD).Read(0xc0001c4880, {0xc002324eb0, 0x9, 0x9})
        	/usr/lib/go-1.18/src/internal/poll/fd_unix.go:167 +0x25a
        net.(*netFD).Read(0xc0001c4880, {0xc002324eb0?, 0x10000005423fc?, 0x7f9e1e119218?})
        	/usr/lib/go-1.18/src/net/fd_posix.go:55 +0x29
        net.(*conn).Read(0xc000010048, {0xc002324eb0?, 0xc002263af8?, 0x44fef2?})
        	/usr/lib/go-1.18/src/net/net.go:183 +0x45
        io.ReadAtLeast({0xff6520, 0xc000010048}, {0xc002324eb0, 0x9, 0x9}, 0x9)
        	/usr/lib/go-1.18/src/io/io.go:331 +0x9a
        io.ReadFull(...)
        	/usr/lib/go-1.18/src/io/io.go:350
        encoding/binary.Read({0xff6520, 0xc000010048}, {0xfff2e8, 0x15972e8}, {0xd3a9e0?, 0xc002324ea0})
        	/usr/lib/go-1.18/src/encoding/binary/binary.go:256 +0xc3e
        github.com/ssbc/go-muxrpc/v2/codec.Reader.ReadHeader({{0xff6520?, 0xc000010048?}}, 0xc002324ea0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/codec/reader.go:48 +0x45
        github.com/ssbc/go-muxrpc/v2.(*Packer).NextHeader(0xc000101e00, {0xffc150, 0xc002348700}, 0xc002324ea0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/packer.go:49 +0xcb
        github.com/ssbc/go-muxrpc/v2.(*rpc).serve.func2(0xc000116410, 0xffc150?, 0xc002263ef8)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:359 +0x5e
        github.com/ssbc/go-muxrpc/v2.(*rpc).serve(0xc000116410)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:378 +0x19c
        github.com/ssbc/go-muxrpc/v2.Handle.func2()
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:128 +0x25
        created by github.com/ssbc/go-muxrpc/v2.Handle
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:127 +0x536
    leakcheck.go:113: Leaked goroutine: goroutine 88 [chan receive]:
        github.com/ssbc/go-muxrpc/v2.(*rpc).Serve(0xc00236a240?)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:336 +0x2b
        github.com/ssbc/go-ssb/sbot.unixSockServer.serve({{0xffc4d0, 0xc0022e25a0}, {0xff5dc0, 0xc00224ccb0}, {0xffb1c0, 0xc0022e2930}, {0x0, 0x0, 0x0}, {0xc0022e2960, ...}, ...}, ...)
        	/home/go-ssb/go-ssb/sbot/options.go:228 +0x4f9
        created by github.com/ssbc/go-ssb/sbot.unixSockServer.accept
        	/home/go-ssb/go-ssb/sbot/options.go:205 +0x51f
    leakcheck.go:113: Leaked goroutine: goroutine 89 [select]:
        context.propagateCancel.func1()
        	/usr/lib/go-1.18/src/context/context.go:279 +0x9e
        created by context.propagateCancel
        	/usr/lib/go-1.18/src/context/context.go:278 +0x1cb
    leakcheck.go:113: Leaked goroutine: goroutine 9 [IO wait]:
        internal/poll.runtime_pollWait(0x7f9deffcf998, 0x72)
        	/usr/lib/go-1.18/src/runtime/netpoll.go:302 +0x89
        internal/poll.(*pollDesc).wait(0xc00015b100?, 0x203000?, 0x0)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:83 +0x32
        internal/poll.(*pollDesc).waitRead(...)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:88
        internal/poll.(*FD).Accept(0xc00015b100)
        	/usr/lib/go-1.18/src/internal/poll/fd_unix.go:614 +0x22c
        net.(*netFD).accept(0xc00015b100)
        	/usr/lib/go-1.18/src/net/fd_unix.go:172 +0x35
        net.(*UnixListener).accept(0x4433c0?)
        	/usr/lib/go-1.18/src/net/unixsock_posix.go:166 +0x1c
        net.(*UnixListener).Accept(0xc0022e2930)
        	/usr/lib/go-1.18/src/net/unixsock.go:260 +0x3d
        github.com/ssbc/go-ssb/sbot.unixSockServer.accept({{0xffc4d0, 0xc0022e25a0}, {0xff5dc0, 0xc00224ccb0}, {0xffb1c0, 0xc0022e2930}, {0x0, 0x0, 0x0}, {0xc0022e2960, ...}, ...})
        	/home/go-ssb/go-ssb/sbot/options.go:175 +0x64
        created by github.com/ssbc/go-ssb/sbot.WithUNIXSocket.func1
        	/home/go-ssb/go-ssb/sbot/options.go:153 +0x33b
    leakcheck.go:113: Leaked goroutine: goroutine 91 [IO wait]:
        internal/poll.runtime_pollWait(0x7f9deffcf7b8, 0x72)
        	/usr/lib/go-1.18/src/runtime/netpoll.go:302 +0x89
        internal/poll.(*pollDesc).wait(0xc0001c4c80?, 0xc002324d90?, 0x0)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:83 +0x32
        internal/poll.(*pollDesc).waitRead(...)
        	/usr/lib/go-1.18/src/internal/poll/fd_poll_runtime.go:88
        internal/poll.(*FD).Read(0xc0001c4c80, {0xc002324d90, 0x9, 0x9})
        	/usr/lib/go-1.18/src/internal/poll/fd_unix.go:167 +0x25a
        net.(*netFD).Read(0xc0001c4c80, {0xc002324d90?, 0x38?, 0x40f845?})
        	/usr/lib/go-1.18/src/net/fd_posix.go:55 +0x29
        net.(*conn).Read(0xc000010060, {0xc002324d90?, 0x7f9e1e10f108?, 0x10?})
        	/usr/lib/go-1.18/src/net/net.go:183 +0x45
        io.ReadAtLeast({0x7f9df4631618, 0xc0001547e0}, {0xc002324d90, 0x9, 0x9}, 0x9)
        	/usr/lib/go-1.18/src/io/io.go:331 +0x9a
        io.ReadFull(...)
        	/usr/lib/go-1.18/src/io/io.go:350
        encoding/binary.Read({0x7f9df4631618, 0xc0001547e0}, {0xfff2e8, 0x15972e8}, {0xd3a9e0?, 0xc002324d80})
        	/usr/lib/go-1.18/src/encoding/binary/binary.go:256 +0xc3e
        github.com/ssbc/go-muxrpc/v2/codec.Reader.ReadHeader({{0x7f9df4631618?, 0xc0001547e0?}}, 0xc002324d80)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/codec/reader.go:48 +0x45
        github.com/ssbc/go-muxrpc/v2.(*Packer).NextHeader(0xc00236a240, {0xffc150, 0xc002348800}, 0xc002324d80)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/packer.go:49 +0xcb
        github.com/ssbc/go-muxrpc/v2.(*rpc).serve.func2(0xc0001165b0, 0xffc150?, 0xc002261ef8)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:359 +0x5e
        github.com/ssbc/go-muxrpc/v2.(*rpc).serve(0xc0001165b0)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:378 +0x19c
        github.com/ssbc/go-muxrpc/v2.Handle.func2()
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:128 +0x25
        created by github.com/ssbc/go-muxrpc/v2.Handle
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:127 +0x536
    leakcheck.go:113: Leaked goroutine: goroutine 94 [chan receive]:
        github.com/ssbc/go-muxrpc/v2.(*rpc).Serve(0xc0001bafd0?)
        	/home/go-ssb/go/pkg/mod/github.com/ssbc/go-muxrpc/v2@v2.0.14-0.20221111190521-10382533750c/rpc_server.go:336 +0x2b
        github.com/ssbc/go-ssb/client.NewUnix.func1()
        	/home/go-ssb/go-ssb/client/client.go:169 +0x4f
        created by github.com/ssbc/go-ssb/client.NewUnix
        	/home/go-ssb/go-ssb/client/client.go:168 +0x425
--- FAIL: TestNames (10.07s)
FAIL
```

--- a/message/publish.go
+++ b/message/publish.go
@@ -29,6 +29,7 @@ type publishLog struct {
 	mu         sync.Mutex
 	byAuthor   margaret.Log
 	receiveLog margaret.Log
+	waitForIndexesCallback func()
 
 	create creater
 }
@@ -57,6 +58,10 @@ func (pl publishLog) Changes() luigi.Observable {
 }
 
 func (pl publishLog) Seq() int64 {
+	if pl.waitForIndexesCallback != nil {
+		pl.waitForIndexesCallback()
+	}
+
 	return pl.byAuthor.Seq()
 }
 
@@ -79,6 +84,11 @@ func (pl publishLog) Query(qry ...margaret.QuerySpec) (luigi.Source, error) {
 }
 
 func (pl *publishLog) Append(val interface{}) (int64, error) {
+	// wait for indexes to catch up before pulling a mutex so we're not locking unnecessarily
+	if pl.waitForIndexesCallback != nil {
+		pl.waitForIndexesCallback()
+	}
+
 	pl.mu.Lock()
 	defer pl.mu.Unlock()
 
@@ -199,6 +209,13 @@ func UseNowTimestamps(yes bool) PublishOption {
 		default:
 			return fmt.Errorf("setTimestamp: unknown creater: %T", cv)
 		}
+		return nil
+	}
+}
+
+func UseWaitForIndexesCallback(cb func()) PublishOption {
+	return func(pl *publishLog) error {
+		pl.waitForIndexesCallback = cb
 		return nil
 	}
 }

--- a/message/publish_test.go
+++ b/message/publish_test.go
@@ -10,6 +10,7 @@ import (
 	"os"
 	"path/filepath"
 	"testing"
+	"time"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
@@ -17,17 +18,11 @@ import (
 	"github.com/ssbc/go-ssb"
 	refs "github.com/ssbc/go-ssb-refs"
 	"github.com/ssbc/go-ssb/internal/asynctesting"
-	"github.com/ssbc/go-ssb/internal/testutils"
 	"github.com/ssbc/go-ssb/multilogs"
 	"github.com/ssbc/go-ssb/repo"
 )
 
 func TestSignMessages(t *testing.T) {
-	if testutils.SkipOnCI(t) {
-		// https://github.com/ssbc/go-ssb/pull/170
-		return
-	}
-
 	tctx := context.TODO()
 	r := require.New(t)
 	a := assert.New(t)
@@ -79,6 +74,11 @@ func TestSignMessages(t *testing.T) {
 		},
 	}
 	for i, msg := range tmsgs {
+		// delay for a bit to allow the indexes to catch up
+		// to be reliable, this has to be done between each publish
+		// TODO: find a way to wait for indexes for this isolated test like we do for sbot
+		time.Sleep(100 * time.Millisecond)
+
 		newSeq, err := w.Append(msg)
 		r.NoError(err, "failed to pour test message %d", i)
 		r.EqualValues(i, newSeq, "advanced")

--- a/sbot/identities.go
+++ b/sbot/identities.go
@@ -22,6 +22,7 @@ func (sbot *Sbot) PublishAs(nick string, val interface{}) (refs.Message, error) 
 
 	var pubopts = []message.PublishOption{
 		message.UseNowTimestamps(true),
+		message.UseWaitForIndexesCallback(sbot.WaitUntilIndexesAreSynced),
 	}
 	if sbot.signHMACsecret != nil { // all feeds use the same settings right now
 		pubopts = append(pubopts, message.SetHMACKey(sbot.signHMACsecret))

--- a/sbot/indexes.go
+++ b/sbot/indexes.go
@@ -222,7 +222,13 @@ func (s *Sbot) serveIndexFrom(name string, snk librarian.SinkIndex, msgs margare
 			s.indexSyncStart()
 		}
 		doneProcessing := func() {
-			s.indexSyncDone()
+			// don't clear the waitgroup right away but give it a little time in case it needs to continue on to another message.
+			// this adds a short (theoretically unnecessary) delay to publish operations but not append operations where
+			// we have received a message from somewhere else.
+			// TODO: eliminate this delay by finding a way to directly check if the luigi queue is completely empty
+			time.AfterFunc(100 * time.Millisecond, func() {
+				s.indexSyncDone()
+			})
 		}
 
 		err = luigi.PumpWithStatus(s.rootCtx, snk, src, startWaiting, doneWaiting, startProcessing, doneProcessing)

--- a/sbot/new.go
+++ b/sbot/new.go
@@ -325,6 +325,7 @@ func New(fopts ...Option) (*Sbot, error) {
 	// publish
 	var pubopts = []message.PublishOption{
 		message.UseNowTimestamps(true),
+		message.UseWaitForIndexesCallback(s.WaitUntilIndexesAreSynced),
 	}
 	if s.signHMACsecret != nil {
 		pubopts = append(pubopts, message.SetHMACKey(s.signHMACsecret))

